// @atlaskit/dynamic-table/dist/cjs/theme.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tableBorder = exports.row = exports.head = exports.arrow = exports.MSThemeColors = void 0;
var colors = _interopRequireWildcard(require("@atlaskit/theme/colors"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
var MSThemeColors = exports.MSThemeColors = {
  Background: 'Canvas',
  Text: 'CanvasText',
  SelectedBackground: 'Highlight',
  SelectedText: 'HighlightText'
};
var arrow = exports.arrow = {
  defaultColor: "var(--ds-icon-disabled, ".concat(colors.N40, ")"),
  selectedColor: "var(--ds-icon-subtle, ".concat(colors.N300, ")")
};
var row = exports.row = {
  focusOutline: "var(--ds-border-focused, ".concat(colors.B100, ")"),
  highlightedBackground: "var(--ds-background-selected, ".concat(colors.B50, ")"),
  hoverBackground: "var(--ds-background-neutral-subtle-hovered, ".concat(colors.N10, ")"),
  hoverHighlightedBackground: "var(--ds-background-selected-hovered, ".concat(colors.B75, ")")
};
var head = exports.head = {
  textColor: "var(--ds-text-subtlest, ".concat(colors.N300, ")")
};
var tableBorder = exports.tableBorder = {
  borderColor: "var(--ds-border, ".concat(colors.N40, ")")
};
