// @atlaskit/pagination/dist/cjs/internal/components/render-ellipsis.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = renderEllipsis;
var _react = _interopRequireDefault(require("react"));
var _primitives = require("@atlaskit/primitives");
var _visuallyHidden = _interopRequireDefault(require("@atlaskit/visually-hidden"));
var containerStyles = (0, _primitives.xcss)({
  display: 'flex',
  position: 'relative'
});
function renderEllipsis(_ref) {
  var key = _ref.key,
    testId = _ref.testId,
    from = _ref.from,
    to = _ref.to;
  return /*#__PURE__*/_react.default.createElement(_primitives.Box, {
    as: "li",
    testId: testId,
    key: key,
    xcss: containerStyles,
    paddingInline: "space.100"
  }, /*#__PURE__*/_react.default.createElement(_primitives.Text, {
    testId: testId && "".concat(testId, "-text")
  }, /*#__PURE__*/_react.default.createElement(_visuallyHidden.default, null, "Skipped pages from ", from, " to ", to), "\u2026"));
}
