// @atlaskit/dynamic-table/dist/cjs/styled/rankable/table-row.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RankableTableBodyRow = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _tableRow = require("../table-row");
var _excluded = ["isRanking", "isRankingItem", "testId"];
/**
 * @jsxRuntime classic
 */
/** @jsx jsx */
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
var rankingStyles = (0, _react2.css)({
  display: 'block'
});
var rankingItemStyles = (0, _react2.css)({
  backgroundColor: "var(--ds-surface-overlay, ".concat(_colors.N20, ")"),
  borderRadius: "var(--ds-border-radius-050, 2px)",
  boxShadow: "var(--ds-shadow-overlay, ".concat("0 20px 32px -8px ".concat(_colors.N50A, ", 0 0 1px ").concat(_colors.N60A), ")")
});
var draggableStyles = (0, _react2.css)({
  outlineWidth: "var(--ds-border-width, 2px)",
  '&:focus-visible': {
    outlineColor: "var(--ds-border-focused, ".concat(_colors.B100, ")"),
    outlineStyle: 'solid'
  }
});

// eslint-disable-next-line @repo/internal/react/require-jsdoc
var RankableTableBodyRow = exports.RankableTableBodyRow = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var isRanking = _ref.isRanking,
    isRankingItem = _ref.isRankingItem,
    testId = _ref.testId,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return (0, _react2.jsx)(_tableRow.TableBodyRow, (0, _extends2.default)({
    css: [isRanking && rankingStyles, isRankingItem && rankingItemStyles, draggableStyles],
    ref: ref,
    testId: testId
  }, props));
});
