// @atlaskit/dynamic-table/dist/cjs/components/loading-container.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _spinner = _interopRequireDefault(require("@atlaskit/spinner"));
var _constants = require("../internal/constants");
var _loadingContainer = require("../styled/loading-container");
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
var LoadingContainer = exports.default = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(LoadingContainer, _React$Component);
  var _super = _createSuper(LoadingContainer);
  function LoadingContainer() {
    (0, _classCallCheck2.default)(this, LoadingContainer);
    return _super.apply(this, arguments);
  }
  (0, _createClass2.default)(LoadingContainer, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        children = _this$props.children,
        isLoading = _this$props.isLoading,
        spinnerSize = _this$props.spinnerSize,
        contentsOpacity = _this$props.contentsOpacity,
        testId = _this$props.testId,
        loadingLabel = _this$props.loadingLabel;
      return /*#__PURE__*/_react.default.createElement(_loadingContainer.Container, {
        testId: testId
      }, !isLoading ? children : /*#__PURE__*/_react.default.createElement(_loadingContainer.ContentsContainer, {
        contentsOpacity: contentsOpacity,
        testId: testId
      }, children), isLoading && /*#__PURE__*/_react.default.createElement(_loadingContainer.SpinnerContainer, {
        testId: testId
      }, /*#__PURE__*/_react.default.createElement(_spinner.default, {
        size: spinnerSize,
        testId: testId && "".concat(testId, "--loadingSpinner"),
        label: loadingLabel
      })));
    }
  }]);
  return LoadingContainer;
}(_react.default.Component);
(0, _defineProperty2.default)(LoadingContainer, "defaultProps", {
  isLoading: true,
  spinnerSize: _constants.LARGE,
  contentsOpacity: _constants.LOADING_CONTENTS_OPACITY,
  loadingLabel: 'Loading table'
});
