// @forge/util/packages/ari/chunk-HNOL6MYE.mjs
import {
  UnifiedHelpRoleAriResourceOwner,
  UnifiedHelpRoleAriResourceType
} from "./chunk-BQV74DFA.mjs";
import {
  RegisteredAri
} from "./chunk-GER6AKWK.mjs";
import {
  AriParser
} from "./chunk-CKE2IRS3.mjs";

// src/unified-help/role/manifest.ts
var unifiedHelpRoleAriStaticOpts = {
  qualifier: "ari",
  platformQualifier: "cloud",
  cloudId: new RegExp("^$"),
  resourceOwner: UnifiedHelpRoleAriResourceOwner,
  resourceType: UnifiedHelpRoleAriResourceType,
  resourceIdSlug: "product/{roleType}",
  resourceIdSegmentFormats: {
    roleType: /(?:member|admin)/
    // eslint-disable-line no-useless-escape
  }
};

// src/unified-help/role/index.ts
var UnifiedHelpRoleAri = class _UnifiedHelpRoleAri extends RegisteredAri {
  constructor(opts) {
    super(opts);
    this._roleType = opts.resourceIdSegmentValues.roleType;
  }
  get roleType() {
    return this._roleType;
  }
  static create(opts) {
    let derivedOpts = {
      qualifier: unifiedHelpRoleAriStaticOpts.qualifier,
      platformQualifier: unifiedHelpRoleAriStaticOpts.platformQualifier,
      cloudId: void 0,
      resourceOwner: unifiedHelpRoleAriStaticOpts.resourceOwner,
      resourceType: unifiedHelpRoleAriStaticOpts.resourceType,
      resourceId: `product/${opts.roleType}`,
      resourceIdSegmentValues: {
        roleType: opts.roleType
      }
    }, ariOpts = AriParser.fromOpts(derivedOpts, unifiedHelpRoleAriStaticOpts);
    return new _UnifiedHelpRoleAri(ariOpts);
  }
  static parse(maybeAri) {
    let opts = AriParser.fromString(maybeAri, unifiedHelpRoleAriStaticOpts);
    return new _UnifiedHelpRoleAri(opts);
  }
  getVariables() {
    return {
      roleType: this.roleType
    };
  }
};

export {
  UnifiedHelpRoleAri
};
