// @forge/util/packages/ari/index.mjs
import "./chunk-NZ5QEVUV.mjs";
import {
  VirtualAgentIntentQuestionProjectionAri
} from "./chunk-4ONMN45A.mjs";
import "./chunk-FRZCZ4T6.mjs";
import {
  VirtualAgentIntentRuleProjectionAri
} from "./chunk-YHBO6CLZ.mjs";
import "./chunk-MJ72T4ZB.mjs";
import {
  VirtualAgentIntentTemplateAri
} from "./chunk-IFVREN5I.mjs";
import "./chunk-66RZPSB4.mjs";
import {
  VirtualAgentConfigurationAri
} from "./chunk-Z6WUEAXN.mjs";
import "./chunk-TPOGUUFE.mjs";
import {
  VirtualAgentConversationAri
} from "./chunk-PEPXK62R.mjs";
import "./chunk-PF3FHNQU.mjs";
import {
  VirtualAgentFlowEditorAri
} from "./chunk-74PPKUSH.mjs";
import "./chunk-ETXO2665.mjs";
import {
  VirtualAgentIntentProjectionAri
} from "./chunk-WBZE3DOB.mjs";
import "./chunk-MPJ7GQCW.mjs";
import "./chunk-ATN7NKOC.mjs";
import {
  UnifiedHelpRoleAri
} from "./chunk-HNOL6MYE.mjs";
import "./chunk-BQV74DFA.mjs";
import {
  UnifiedHelpSiteAri
} from "./chunk-BSZAH2HZ.mjs";
import "./chunk-OKKAA5RV.mjs";
import {
  UnifiedHelpWorkspaceAri
} from "./chunk-ZGO54XL6.mjs";
import "./chunk-GXNUL5AQ.mjs";
import "./chunk-4MWZBX5W.mjs";
import {
  TrelloRoleAri
} from "./chunk-YRCFSGZA.mjs";
import "./chunk-I5GECIAC.mjs";
import {
  TrelloSiteAri
} from "./chunk-Y2UGON3S.mjs";
import "./chunk-P3FZABZE.mjs";
import {
  TrelloUserAri
} from "./chunk-PFMWGUK7.mjs";
import "./chunk-OZFY5Y6K.mjs";
import {
  TrelloWorkspaceAri
} from "./chunk-7QMFZ5TB.mjs";
import "./chunk-KVD2HPFJ.mjs";
import {
  TrelloBoardAri
} from "./chunk-UPC4CAS4.mjs";
import "./chunk-HI4UEG2F.mjs";
import {
  TrelloCardAri
} from "./chunk-TWRNEYUX.mjs";
import "./chunk-L5R7QT5M.mjs";
import {
  TrelloEnterpriseAri
} from "./chunk-SPIF2CSS.mjs";
import "./chunk-A3QGWDCS.mjs";
import {
  TrelloListAri
} from "./chunk-DBVVE2JK.mjs";
import "./chunk-TVSKQH7F.mjs";
import "./chunk-D3TDYNKN.mjs";
import {
  TownsquareQuestionAri
} from "./chunk-NI7NDIOX.mjs";
import "./chunk-IKIQRIWE.mjs";
import {
  TownsquareRoleAri
} from "./chunk-H6JDSFX7.mjs";
import "./chunk-RRNPSY4J.mjs";
import {
  TownsquareSiteAri
} from "./chunk-IEH23PLB.mjs";
import "./chunk-MC7VKTYC.mjs";
import {
  TownsquareTagAri
} from "./chunk-XFA7UJ2L.mjs";
import "./chunk-7QVDZS4D.mjs";
import {
  TownsquareHelpPointerAri
} from "./chunk-N5LCF2BQ.mjs";
import "./chunk-GROHW3KB.mjs";
import {
  TownsquareLearningAri
} from "./chunk-IUMSEBRR.mjs";
import "./chunk-4J5O222V.mjs";
import {
  TownsquareProjectAri
} from "./chunk-2I6WZKAQ.mjs";
import "./chunk-ICBJW6N7.mjs";
import "./chunk-LERUK3EA.mjs";
import "./chunk-7BIQUOJP.mjs";
import {
  TeamsTeamAri
} from "./chunk-3HPD37GN.mjs";
import "./chunk-MUICIWTK.mjs";
import {
  TownsquareCommentAri
} from "./chunk-IF6T4TBC.mjs";
import "./chunk-GVAPRDNZ.mjs";
import {
  TownsquareGoalAri
} from "./chunk-C7M5NH7X.mjs";
import "./chunk-FPGL2PN7.mjs";
import "./chunk-2EFBVLUL.mjs";
import "./chunk-TYQPBEKA.mjs";
import {
  StatuspageRoleAri
} from "./chunk-WQYA5F6D.mjs";
import "./chunk-BT2GYN5W.mjs";
import {
  StatuspageSiteAri
} from "./chunk-EVAJJQH3.mjs";
import "./chunk-6IR7P2SS.mjs";
import {
  SupportCustomerAri
} from "./chunk-P2CYNIB3.mjs";
import "./chunk-PHBXUJAW.mjs";
import "./chunk-NTJ25HLA.mjs";
import {
  RuntimeAuthClientRoleAri
} from "./chunk-Z6MWXQQ6.mjs";
import "./chunk-N3DYIPHN.mjs";
import {
  SearchEventAri
} from "./chunk-76K3EVMF.mjs";
import "./chunk-MLTMULJ4.mjs";
import "./chunk-27WJ432L.mjs";
import {
  PollinatorCheckAri
} from "./chunk-YEBGY5XI.mjs";
import "./chunk-M6X7UPKX.mjs";
import "./chunk-XFIQH5HF.mjs";
import {
  PostOfficeMessageTemplateAri
} from "./chunk-KK5COTT5.mjs";
import "./chunk-YT2S6YE6.mjs";
import {
  PostOfficeMessageInstanceAri
} from "./chunk-QFRHSAGS.mjs";
import "./chunk-6N62CYFU.mjs";
import "./chunk-EHBNQJ47.mjs";
import {
  PlatformSiteAri
} from "./chunk-RKYXNW4C.mjs";
import "./chunk-QGU7KDYS.mjs";
import {
  PlatformLifecycleResourceAri
} from "./chunk-EQRQSSYW.mjs";
import "./chunk-T36754GS.mjs";
import {
  PlatformOrgUserAri
} from "./chunk-LELT2IR5.mjs";
import "./chunk-REH6WMJF.mjs";
import {
  PlatformOrgAri
} from "./chunk-IB3GIWM6.mjs";
import "./chunk-ON6PRZUL.mjs";
import {
  PlatformSecureTunnelAri
} from "./chunk-LBLGBN2Q.mjs";
import "./chunk-7AEMSYL6.mjs";
import {
  PlatformClassificationTagAri
} from "./chunk-FJ44J54V.mjs";
import "./chunk-JOF4WRFT.mjs";
import {
  PlatformLifecycleResourcePackageTypeAri
} from "./chunk-SCUTZX26.mjs";
import "./chunk-YUROJJWW.mjs";
import {
  PlatformLifecycleResourcePackageAri
} from "./chunk-XG7IVVXF.mjs";
import "./chunk-AUDUMQTZ.mjs";
import "./chunk-WL6TPSYP.mjs";
import {
  PeoplePerftoolCalibrationAri
} from "./chunk-KHUWPOGM.mjs";
import "./chunk-YFDO5OIQ.mjs";
import {
  PeoplePerftoolFeedbackAri
} from "./chunk-V3DVBUMU.mjs";
import "./chunk-BUB6ILHZ.mjs";
import {
  PeoplePerftoolPerformanceAri
} from "./chunk-V2HRYEEE.mjs";
import "./chunk-BBPB2HIH.mjs";
import "./chunk-MGHXHAL5.mjs";
import {
  PlatformServicesStreamhubSchemaAri
} from "./chunk-XAA56QX2.mjs";
import "./chunk-KM3YDOZJ.mjs";
import "./chunk-P6ODAFPN.mjs";
import {
  PapiApiAri
} from "./chunk-HKV2UVXB.mjs";
import "./chunk-5IDHPJZF.mjs";
import {
  PapiRoleAri
} from "./chunk-2DEG6W7D.mjs";
import "./chunk-NLJJ6GJ4.mjs";
import "./chunk-2K5F5KXA.mjs";
import {
  PassionfruitUserAri
} from "./chunk-VOUUTHLU.mjs";
import "./chunk-Q2W2OS63.mjs";
import "./chunk-UW7FK2QY.mjs";
import {
  OpsgenieSiteAri
} from "./chunk-KO4PNN5D.mjs";
import "./chunk-3A7NUHDU.mjs";
import {
  OpsgenieTeamAri
} from "./chunk-HH2TJ5JA.mjs";
import "./chunk-5GDLDNUH.mjs";
import {
  OpsgenieTimelineAri
} from "./chunk-7MJ4NWWB.mjs";
import "./chunk-T4JHTCBL.mjs";
import {
  OpsgenieWorkspaceAri
} from "./chunk-OSROIG42.mjs";
import "./chunk-Y2SQLTAK.mjs";
import {
  OpsgenieNotificationAri
} from "./chunk-IR3UKK3W.mjs";
import "./chunk-WBXUDE5Y.mjs";
import {
  OpsgenieRoleAri
} from "./chunk-TGHIDHGQ.mjs";
import "./chunk-7ZPWMXWE.mjs";
import {
  OpsgenieScheduleRotationAri
} from "./chunk-KQD3JFUT.mjs";
import "./chunk-ILPXSBCV.mjs";
import {
  OpsgenieScheduleAri
} from "./chunk-PCP2FVS5.mjs";
import "./chunk-OFBJZXGI.mjs";
import {
  OpsgenieIncidentAri
} from "./chunk-7GZNWK7E.mjs";
import "./chunk-TLNXXINJ.mjs";
import {
  OpsgenieIncomingCallHistoryAri
} from "./chunk-D6ZK23QS.mjs";
import "./chunk-W4G5P22J.mjs";
import {
  OpsgenieIntegrationAri
} from "./chunk-T5EIOYUX.mjs";
import "./chunk-KNRSSKUB.mjs";
import {
  OpsgenieEscalationAri
} from "./chunk-7VVBVRY2.mjs";
import "./chunk-C3Q7ZVPL.mjs";
import {
  OpsgenieEventAri
} from "./chunk-JVCU3BHX.mjs";
import "./chunk-PW5GVXRG.mjs";
import {
  OpsgenieIncidentAlertLinkAri
} from "./chunk-BHIY7FTB.mjs";
import "./chunk-JCLOAQNZ.mjs";
import {
  OpsgenieIncidentStatusUpdateAri
} from "./chunk-I7S5SSWS.mjs";
import "./chunk-RO4YS42G.mjs";
import {
  OpsgenieCallRoutingAri
} from "./chunk-FQIJHWIL.mjs";
import "./chunk-P2OBBGSQ.mjs";
import {
  OpsgenieCustomRoleAri
} from "./chunk-IJJCTIWR.mjs";
import "./chunk-RW54AUFT.mjs";
import {
  OpsgenieDeploymentServiceLinkAri
} from "./chunk-7MJSDMYV.mjs";
import "./chunk-VYEUWCEP.mjs";
import {
  OpsgenieDeploymentAri
} from "./chunk-OBBP3JRA.mjs";
import "./chunk-BVX743QJ.mjs";
import {
  OpsgenieAccountSettingsAri
} from "./chunk-KHPYXCHM.mjs";
import "./chunk-OZDNDKVC.mjs";
import {
  OpsgenieAlertRecipientLinkAri
} from "./chunk-VCJ7MZ5S.mjs";
import "./chunk-JXI5TFG4.mjs";
import {
  OpsgenieAlertAri
} from "./chunk-BGLDYMUW.mjs";
import "./chunk-PZKFI7JM.mjs";
import {
  OpsgenieApiRequestMetricAri
} from "./chunk-YRMBQXN4.mjs";
import "./chunk-R4SGT2SX.mjs";
import "./chunk-IXJTRB7L.mjs";
import {
  OauthClientAri
} from "./chunk-MN7T4T25.mjs";
import "./chunk-SLRY2ZFP.mjs";
import {
  OpsgenieAccountLoginAri
} from "./chunk-2A7XMAI2.mjs";
import "./chunk-B2UP53JZ.mjs";
import "./chunk-DU3KZ7XN.mjs";
import "./chunk-SV3OSS6S.mjs";
import {
  MercuryRoleAri
} from "./chunk-ZNN4C6YW.mjs";
import "./chunk-2CIDR5KT.mjs";
import {
  MercurySiteAri
} from "./chunk-UXIVCYBT.mjs";
import "./chunk-SEACGWCF.mjs";
import {
  MercuryCommentAri
} from "./chunk-HODDT74C.mjs";
import "./chunk-IP7NNLNW.mjs";
import {
  MercuryProgramStatusUpdateAri
} from "./chunk-X2CQ5AXE.mjs";
import "./chunk-DXGTR4X7.mjs";
import {
  MercuryProgramAri
} from "./chunk-4FQQSYO4.mjs";
import "./chunk-YNUAOINR.mjs";
import "./chunk-ZNQVUOOQ.mjs";
import {
  MeasurementSiteUserAri
} from "./chunk-NTZLP4CJ.mjs";
import "./chunk-CHIUNEWO.mjs";
import {
  MeasurementUserAri
} from "./chunk-HL37J4CE.mjs";
import "./chunk-G4CB4RQJ.mjs";
import {
  MediaFileAri
} from "./chunk-CVA5UJ6N.mjs";
import "./chunk-D3IDOZFP.mjs";
import "./chunk-ZM3JBOHH.mjs";
import "./chunk-ITJPM4K6.mjs";
import {
  MarketingCustomerDomainAri
} from "./chunk-CBCUXP7A.mjs";
import "./chunk-VAIM2B7I.mjs";
import "./chunk-JEDPRILO.mjs";
import {
  MarketplaceTeamAri
} from "./chunk-7W33A7UX.mjs";
import "./chunk-IUUUDGF6.mjs";
import {
  MeasurementEmailUuidAri
} from "./chunk-LJJQW7HN.mjs";
import "./chunk-TZWB442U.mjs";
import "./chunk-Y2UB4MCB.mjs";
import {
  LinkingPlatformDatasourceAri
} from "./chunk-N64GDVNS.mjs";
import "./chunk-ARLPVCER.mjs";
import {
  MakerSpaceGroupAri
} from "./chunk-GNTTN4VA.mjs";
import "./chunk-TI7RQAMT.mjs";
import "./chunk-XTCV6VAZ.mjs";
import {
  JiraWorkspaceAri
} from "./chunk-ACG2FM3K.mjs";
import "./chunk-KOJWXOCJ.mjs";
import {
  JiraVulnerabilityAri
} from "./chunk-BXFPGADP.mjs";
import "./chunk-2NDER42Y.mjs";
import {
  JiraWorkflowSchemeAri
} from "./chunk-TCBWCW3Q.mjs";
import "./chunk-GF5WTUWI.mjs";
import {
  JiraWorkflowAri
} from "./chunk-Y52VBLHT.mjs";
import "./chunk-EYJJDCXK.mjs";
import {
  JiraWorklogAri
} from "./chunk-OAXEFG37.mjs";
import "./chunk-V5CUEKO2.mjs";
import {
  JiraUserBroadcastMessageAri
} from "./chunk-RCZIOTO3.mjs";
import "./chunk-MHNZKONH.mjs";
import {
  JiraVersionApproverAri
} from "./chunk-DW3MPW5T.mjs";
import "./chunk-CLH2DEO3.mjs";
import {
  JiraVersionAri
} from "./chunk-KW2YBEO4.mjs";
import "./chunk-LN36NVHR.mjs";
import {
  JiraVulnerabilityHistoryAri
} from "./chunk-IEO7TH2W.mjs";
import "./chunk-DQSHJLUH.mjs";
import {
  JiraSecurityLevelAri
} from "./chunk-YTAYQ5N5.mjs";
import "./chunk-PO2YLXFH.mjs";
import {
  JiraSecurityWorkspaceAri
} from "./chunk-6HNA6AF7.mjs";
import "./chunk-7PTBO36C.mjs";
import {
  JiraSiteAri
} from "./chunk-7XNITNOV.mjs";
import "./chunk-XIFA6EHM.mjs";
import {
  JiraSprintAri
} from "./chunk-XZCHOM6T.mjs";
import "./chunk-FGZEBJS7.mjs";
import {
  JiraResourceUsageMetricAri
} from "./chunk-RCPQXGXQ.mjs";
import "./chunk-LLLNN75K.mjs";
import {
  JiraResourceUsageRecommendationAri
} from "./chunk-YUWR5QKV.mjs";
import "./chunk-D7TMYPCS.mjs";
import {
  JiraRoleAri
} from "./chunk-I63UJYVE.mjs";
import "./chunk-WAQ2TE25.mjs";
import {
  JiraSecurityContainerAri
} from "./chunk-25DAELO3.mjs";
import "./chunk-4RYB4RHN.mjs";
import {
  JiraRemoteLinkHistoryAri
} from "./chunk-B5QDKM3T.mjs";
import "./chunk-QY36TOVX.mjs";
import {
  JiraRemoteLinkAri
} from "./chunk-NJ5GVNVX.mjs";
import "./chunk-KHCVG4TJ.mjs";
import {
  JiraRepositoryAri
} from "./chunk-ILBTNVYF.mjs";
import "./chunk-PYOC3DJR.mjs";
import {
  JiraResolutionAri
} from "./chunk-JQ6PILYV.mjs";
import "./chunk-BOQOC7ST.mjs";
import {
  JiraProjectTypeAri
} from "./chunk-PFX7RB5U.mjs";
import "./chunk-OF3K3RDZ.mjs";
import {
  JiraProjectAri
} from "./chunk-QKYD2JKG.mjs";
import "./chunk-CQAE6WHE.mjs";
import {
  JiraPullRequestHistoryAri
} from "./chunk-JDLPKZ6O.mjs";
import "./chunk-IBNDOKW7.mjs";
import {
  JiraPullRequestAri
} from "./chunk-Q5CKRKTF.mjs";
import "./chunk-5FKWU5BE.mjs";
import {
  JiraProjectFeatureAri
} from "./chunk-IVJUFPLC.mjs";
import "./chunk-QKR33HHC.mjs";
import {
  JiraProjectOverviewAri
} from "./chunk-35LIYFFV.mjs";
import "./chunk-ORHZXN2K.mjs";
import {
  JiraProjectRoleActorAri
} from "./chunk-ADFKPCKA.mjs";
import "./chunk-6XVY5ZJP.mjs";
import {
  JiraProjectShortcutAri
} from "./chunk-Q6346KCA.mjs";
import "./chunk-MAZNZE5K.mjs";
import {
  JiraPostIncidentReviewAri
} from "./chunk-H3OEJNK4.mjs";
import "./chunk-D5S55TXO.mjs";
import {
  JiraPriorityAri
} from "./chunk-EWKXLKQH.mjs";
import "./chunk-C4M7DMM2.mjs";
import {
  JiraProductAri
} from "./chunk-ONMEIEV6.mjs";
import "./chunk-DZZ6Y4HB.mjs";
import {
  JiraProjectCategoryAri
} from "./chunk-4MOCOH55.mjs";
import "./chunk-23L4LUZM.mjs";
import {
  JiraPermissionSchemeAri
} from "./chunk-62FHAADP.mjs";
import "./chunk-SASLBDN4.mjs";
import {
  JiraPlanAri
} from "./chunk-OW6TNHKG.mjs";
import "./chunk-GGTGTA2K.mjs";
import {
  JiraPostIncidentReviewHistoryAri
} from "./chunk-RNL3YZ4T.mjs";
import "./chunk-XL2VWTF2.mjs";
import {
  JiraPostIncidentReviewLinkAri
} from "./chunk-GHHUNUAG.mjs";
import "./chunk-WSTI2GHH.mjs";
import {
  JiraNotificationTypeSchemeAri
} from "./chunk-OTFZVYIJ.mjs";
import "./chunk-45SACCRI.mjs";
import {
  JiraNotificationTypeAri
} from "./chunk-U64HSRRM.mjs";
import "./chunk-QXA66ZDW.mjs";
import {
  JiraNotificationUserPreferenceAri
} from "./chunk-QMCA6NEX.mjs";
import "./chunk-RYGO66NC.mjs";
import {
  JiraOperationsWorkspaceAri
} from "./chunk-SQZWDGZM.mjs";
import "./chunk-QTWVEF4M.mjs";
import {
  JiraIssueWorklogAri
} from "./chunk-HE2W7JOW.mjs";
import "./chunk-O6FNFDKY.mjs";
import {
  JiraIssueAri
} from "./chunk-GJLRAHJ2.mjs";
import "./chunk-6MDCCCXR.mjs";
import {
  JiraIssuefieldvalueAri
} from "./chunk-3LEALWCG.mjs";
import "./chunk-ZNA5VEMQ.mjs";
import {
  JiraNavigationItemAri
} from "./chunk-3I7346J3.mjs";
import "./chunk-GXFJTOBO.mjs";
import {
  JiraIssueStatusAri
} from "./chunk-4ZTGV4PA.mjs";
import "./chunk-S3WTNWJL.mjs";
import {
  JiraIssueTypeSchemeMappingAri
} from "./chunk-MULWN4FE.mjs";
import "./chunk-VNWAD3NX.mjs";
import {
  JiraIssueTypeSchemeAri
} from "./chunk-3DXIZ4CC.mjs";
import "./chunk-QFTRTL74.mjs";
import {
  JiraIssueTypeAri
} from "./chunk-HT5F3G3S.mjs";
import "./chunk-YBV4ZT7H.mjs";
import {
  JiraIssueLinkTypeAri
} from "./chunk-BQBHVB3T.mjs";
import "./chunk-ZJWWJNVZ.mjs";
import {
  JiraIssueLinkAri
} from "./chunk-QFBXKPBB.mjs";
import "./chunk-QRQQNGKX.mjs";
import {
  JiraIssueRemoteLinkAri
} from "./chunk-A5PV7UFX.mjs";
import "./chunk-QO2BQQRK.mjs";
import {
  JiraIssueSearchViewAri
} from "./chunk-UKLMXEUM.mjs";
import "./chunk-4BHYFYKX.mjs";
import {
  JiraIssueCommentAri
} from "./chunk-NCFOTZ5E.mjs";
import "./chunk-C7LAHHR5.mjs";
import {
  JiraIssueFieldMetadataAri
} from "./chunk-RVKEJTHZ.mjs";
import "./chunk-Q3PRSOZC.mjs";
import {
  JiraIssueFieldOptionAri
} from "./chunk-MI45WCOD.mjs";
import "./chunk-TKZ6M5JE.mjs";
import {
  JiraIssueHistoryAri
} from "./chunk-LTKC2RDW.mjs";
import "./chunk-N64LAJN2.mjs";
import {
  JiraFilterAri
} from "./chunk-4GFGH427.mjs";
import "./chunk-NQBZEVCS.mjs";
import {
  JiraIncidentHistoryAri
} from "./chunk-GGZ6B5FF.mjs";
import "./chunk-7AJWNJ45.mjs";
import {
  JiraIncidentAri
} from "./chunk-JNHBZI3A.mjs";
import "./chunk-5F3IR2Z7.mjs";
import {
  JiraIssueAttachmentAri
} from "./chunk-Y6VL3KC2.mjs";
import "./chunk-AU74CHJB.mjs";
import {
  JiraFavouriteAri
} from "./chunk-PVRHJUY4.mjs";
import "./chunk-ML5L5N3V.mjs";
import {
  JiraFeatureFlagHistoryAri
} from "./chunk-5ODQQXOT.mjs";
import "./chunk-BLEHWG53.mjs";
import {
  JiraFeatureFlagAri
} from "./chunk-EVNKSIFL.mjs";
import "./chunk-KSD4G6QQ.mjs";
import {
  JiraFilterEmailSubscriptionAri
} from "./chunk-H4TJ4TXP.mjs";
import "./chunk-Z6KRUHKY.mjs";
import {
  JiraDevopsComponentAri
} from "./chunk-P2BOCMBJ.mjs";
import "./chunk-7UM4GJSV.mjs";
import {
  JiraDocumentHistoryAri
} from "./chunk-7Z7IHHGL.mjs";
import "./chunk-Z2JLOUYO.mjs";
import {
  JiraDocumentAri
} from "./chunk-CIDRH7EK.mjs";
import "./chunk-LUBY6O72.mjs";
import {
  JiraEntityPropertyAri
} from "./chunk-LY26SELY.mjs";
import "./chunk-WVAVMODN.mjs";
import {
  JiraDeploymentHistoryAri
} from "./chunk-3E44VMHL.mjs";
import "./chunk-UD7ZCFKR.mjs";
import {
  JiraDeploymentAri
} from "./chunk-Y4GU2CUN.mjs";
import "./chunk-ACU6GJRH.mjs";
import {
  JiraDesignHistoryAri
} from "./chunk-RSVRH53W.mjs";
import "./chunk-UG2AH2TA.mjs";
import {
  JiraDesignAri
} from "./chunk-IVSSLHGL.mjs";
import "./chunk-DSHHZHIG.mjs";
import {
  JiraCommitAri
} from "./chunk-TGH25CD5.mjs";
import "./chunk-YBNRWWV7.mjs";
import {
  JiraComponentAri
} from "./chunk-ZZ267JDR.mjs";
import "./chunk-DUSRJMGF.mjs";
import {
  JiraConnectAppAri
} from "./chunk-UW66NU3X.mjs";
import "./chunk-VRZRWJMV.mjs";
import {
  JiraDashboardAri
} from "./chunk-5WKCOB6A.mjs";
import "./chunk-KZLF4LHF.mjs";
import {
  JiraBuildHistoryAri
} from "./chunk-TUXVVM5E.mjs";
import "./chunk-55Z3USGX.mjs";
import {
  JiraBuildAri
} from "./chunk-BXQBMUAF.mjs";
import "./chunk-I56NWCQF.mjs";
import {
  JiraBulkOperationTaskAri
} from "./chunk-PKE23AYL.mjs";
import "./chunk-Z76VAYJY.mjs";
import {
  JiraCommentAri
} from "./chunk-KGYLSEEF.mjs";
import "./chunk-UWONVETB.mjs";
import {
  JiraAnnouncementBannerAri
} from "./chunk-RT27YURI.mjs";
import "./chunk-VFZPJ7ZD.mjs";
import {
  JiraAvailableFieldsAri
} from "./chunk-E4FHOAWC.mjs";
import "./chunk-ICE3OR6G.mjs";
import {
  JiraBranchHistoryAri
} from "./chunk-PP3FEP76.mjs";
import "./chunk-LZMQ2B5J.mjs";
import {
  JiraBranchAri
} from "./chunk-P4TYXXYN.mjs";
import "./chunk-65K6V5WX.mjs";
import "./chunk-NWN6AJRD.mjs";
import {
  JiraSoftwareSiteAri
} from "./chunk-HSPNOZIA.mjs";
import "./chunk-CKKWVELS.mjs";
import {
  JiraSoftwareSwimlaneAri
} from "./chunk-T2HYTFK6.mjs";
import "./chunk-IATVHO4Z.mjs";
import {
  JiraSoftwareUserBoardConfigAri
} from "./chunk-YBHVMB4U.mjs";
import "./chunk-3UUDXEXS.mjs";
import {
  JiraSoftwareViewConfigAri
} from "./chunk-OI5MJ6OI.mjs";
import "./chunk-ZOFOPGOV.mjs";
import {
  JiraSoftwareColumnAri
} from "./chunk-ZKFE5DJN.mjs";
import "./chunk-P7LHCIIY.mjs";
import {
  JiraSoftwareCustomFilterAri
} from "./chunk-XTOG2PX6.mjs";
import "./chunk-O6OVDRBO.mjs";
import {
  JiraSoftwareRoleAri
} from "./chunk-JXJMVDNR.mjs";
import "./chunk-SKSGTSPT.mjs";
import {
  JiraSoftwareCardColorAri
} from "./chunk-GSNFLROX.mjs";
import "./chunk-OWEX3L44.mjs";
import {
  JiraSoftwareCardLayoutAri
} from "./chunk-5T3TDDBU.mjs";
import "./chunk-QSH6F5SY.mjs";
import {
  JiraSoftwareCardParentAri
} from "./chunk-MSFCV7YV.mjs";
import "./chunk-M7R3FF5D.mjs";
import {
  JiraSoftwareCardAri
} from "./chunk-43B3RTNH.mjs";
import "./chunk-6WK3XU6H.mjs";
import {
  JiraSoftwareBoardFeatureAri
} from "./chunk-RMWUIIKA.mjs";
import "./chunk-E24RP2LJ.mjs";
import {
  JiraSoftwareBoardIssueListAri
} from "./chunk-Y233WLCW.mjs";
import "./chunk-3Z6ZDJ2A.mjs";
import {
  JiraSoftwareBoardAri
} from "./chunk-43PQOG5Q.mjs";
import "./chunk-YJAJS2GF.mjs";
import "./chunk-WWZKXHXC.mjs";
import {
  JiraServicedeskSlaAri
} from "./chunk-RKOVDBZN.mjs";
import "./chunk-BMLW67PZ.mjs";
import {
  JiraServicedeskQueueAri
} from "./chunk-PLRXC56E.mjs";
import "./chunk-TC6T4VFX.mjs";
import {
  JiraServicedeskRequestTypeAri
} from "./chunk-VKTGAXQO.mjs";
import "./chunk-5SWOR5KD.mjs";
import {
  JiraServicedeskRoleAri
} from "./chunk-LX22A2K7.mjs";
import "./chunk-VV22WWWI.mjs";
import {
  JiraServicedeskSiteAri
} from "./chunk-UKKTHUQR.mjs";
import "./chunk-F4LW6XYO.mjs";
import {
  JiraServicedeskApprovalAri
} from "./chunk-3AQPZPLH.mjs";
import "./chunk-AZJHCKNE.mjs";
import {
  JiraServicedeskCalendarEventAri
} from "./chunk-SGUPSLOB.mjs";
import "./chunk-SMZUBZEM.mjs";
import {
  JiraServicedeskCannedResponseAri
} from "./chunk-ZDQ6UIMU.mjs";
import "./chunk-XGGNYA2D.mjs";
import {
  JiraServicedeskOrganizationAri
} from "./chunk-QSKHNSJY.mjs";
import "./chunk-FYLFP7EF.mjs";
import "./chunk-P2RGZTM3.mjs";
import {
  JiraAlignInstanceAri
} from "./chunk-BXUTI3E5.mjs";
import "./chunk-KIGWQUKL.mjs";
import "./chunk-QAQMSQTV.mjs";
import {
  JiraCoreRoleAri
} from "./chunk-TP5HAUYJ.mjs";
import "./chunk-22MNSNE6.mjs";
import {
  JiraCoreSiteAri
} from "./chunk-V3I6LVZN.mjs";
import "./chunk-BYLK2PHJ.mjs";
import {
  JiraAlignEpicAri
} from "./chunk-YNFP36JR.mjs";
import "./chunk-5DRIX3CB.mjs";
import "./chunk-7ZZEN3W6.mjs";
import {
  IdentityUserGrantAri
} from "./chunk-F4TTFDSU.mjs";
import "./chunk-7Y76725A.mjs";
import {
  IdentityUserbaseAri
} from "./chunk-HMEA6F2K.mjs";
import "./chunk-FKBB466V.mjs";
import {
  IdentitySiteAri
} from "./chunk-CMZBW53D.mjs";
import "./chunk-CXXH2SWL.mjs";
import {
  IdentityTeamMemberAri
} from "./chunk-JUPUWL3E.mjs";
import "./chunk-OULHQOIR.mjs";
import {
  IdentityTeamAri
} from "./chunk-Q53COVPY.mjs";
import "./chunk-T76PG5FI.mjs";
import {
  IdentityUserAri
} from "./chunk-DYXISX5W.mjs";
import "./chunk-N5QB45QP.mjs";
import {
  IdentityDomainAri
} from "./chunk-V3INIZLC.mjs";
import "./chunk-S5UAML7R.mjs";
import {
  IdentityGroupAri
} from "./chunk-3U2SCTIF.mjs";
import "./chunk-IZPJNHZU.mjs";
import {
  IdentityOauthClientAri
} from "./chunk-EYBREKCA.mjs";
import "./chunk-UTUDGZLV.mjs";
import {
  IdentityRoleAri
} from "./chunk-R6WKFBRO.mjs";
import "./chunk-UFIIDAET.mjs";
import {
  IdentityAuthPolicyAri
} from "./chunk-W7G5AGLI.mjs";
import "./chunk-27IAW7JZ.mjs";
import {
  IdentityCustomerDirectoryAri
} from "./chunk-NACMSOGY.mjs";
import "./chunk-KQGM5YYP.mjs";
import {
  IdentityCustomerOrganizationAri
} from "./chunk-XKAMX47F.mjs";
import "./chunk-XIBIRHB4.mjs";
import {
  IdentityDirectoryAri
} from "./chunk-DHSKEKEY.mjs";
import "./chunk-L36L7XSX.mjs";
import {
  IdentityAppAri
} from "./chunk-KF4H7ZNZ.mjs";
import "./chunk-QULKIQO5.mjs";
import "./chunk-BQGCFS74.mjs";
import {
  HelpLayoutAri
} from "./chunk-4UDUIJ6L.mjs";
import "./chunk-HXZGYRZT.mjs";
import {
  HelpPortalAri
} from "./chunk-RZBU67ZQ.mjs";
import "./chunk-DEFOWKTU.mjs";
import {
  HelpRequestFormAri
} from "./chunk-JRX3DCMK.mjs";
import "./chunk-QF6HVRWR.mjs";
import {
  HelpChannelAri
} from "./chunk-AAB5S5N7.mjs";
import "./chunk-H2XK65KU.mjs";
import {
  HelpHelpCenterAri
} from "./chunk-WIVZBGP4.mjs";
import "./chunk-UIZK5L5W.mjs";
import {
  HelpHelpDeskAri
} from "./chunk-PVMNXSMU.mjs";
import "./chunk-EQROJBZ3.mjs";
import {
  HelpArticleAri
} from "./chunk-BKG3PIAT.mjs";
import "./chunk-RDV7AM67.mjs";
import "./chunk-BRXMD5AP.mjs";
import {
  GraphVulnerabilityHistoryAri
} from "./chunk-2C2DFNCC.mjs";
import "./chunk-PHKBIUQV.mjs";
import {
  GraphVulnerabilityAri
} from "./chunk-7UVQTEFF.mjs";
import "./chunk-J7VD66AE.mjs";
import {
  GraphWorkspaceAri
} from "./chunk-ICL4R434.mjs";
import "./chunk-SQM2NR5M.mjs";
import {
  GraphServiceAndOperationsTeamRelationshipAri
} from "./chunk-GN5XK3PQ.mjs";
import "./chunk-E4XM6JBY.mjs";
import {
  GraphServiceAndVcsRepositoryRelationshipAri
} from "./chunk-XLZ44XHZ.mjs";
import "./chunk-SYSNNBBE.mjs";
import {
  GraphServiceRelationshipAri
} from "./chunk-5563EXS2.mjs";
import "./chunk-JNGK7AYM.mjs";
import {
  GraphServiceAri
} from "./chunk-ECMMO6EV.mjs";
import "./chunk-KKDL7RK7.mjs";
import {
  GraphRepositoryAri
} from "./chunk-VV6K4UD5.mjs";
import "./chunk-SAJO4IGE.mjs";
import {
  GraphSecurityContainerAri
} from "./chunk-SYE4D56Y.mjs";
import "./chunk-6HYQV6SW.mjs";
import {
  GraphSecurityWorkspaceAri
} from "./chunk-PRD3ROSS.mjs";
import "./chunk-4FPKBG6M.mjs";
import {
  GraphServiceAndJiraProjectRelationshipAri
} from "./chunk-7RS37377.mjs";
import "./chunk-723KYNJS.mjs";
import {
  GraphPullRequestHistoryAri
} from "./chunk-K6UH3CTU.mjs";
import "./chunk-K4SS2RGK.mjs";
import {
  GraphPullRequestAri
} from "./chunk-M4IM2NFZ.mjs";
import "./chunk-Z3FWECDS.mjs";
import {
  GraphRemoteLinkHistoryAri
} from "./chunk-I2MQ6GSJ.mjs";
import "./chunk-2YKXM3UN.mjs";
import {
  GraphRemoteLinkAri
} from "./chunk-2MS54QUP.mjs";
import "./chunk-7CH7D4HJ.mjs";
import {
  GraphJiraProjectAndVcsRepositoryRelationshipAri
} from "./chunk-P47SL6JR.mjs";
import "./chunk-XYDSJQO4.mjs";
import {
  GraphOperationsWorkspaceAri
} from "./chunk-KQID34SN.mjs";
import "./chunk-5HHNCVXF.mjs";
import {
  GraphPostIncidentReviewHistoryAri
} from "./chunk-UETLATZU.mjs";
import "./chunk-PVBPUKXI.mjs";
import {
  GraphPostIncidentReviewAri
} from "./chunk-5K4PEHFO.mjs";
import "./chunk-SNAPC4SA.mjs";
import {
  GraphJiraProjectAndDevopsToolRelationshipAri
} from "./chunk-ERRTTRYZ.mjs";
import "./chunk-2A255S76.mjs";
import {
  GraphJiraProjectAndDocumentationSpaceRelationshipAri
} from "./chunk-TRMBOR4U.mjs";
import "./chunk-SM4OYQAG.mjs";
import {
  GraphJiraProjectAndOperationsTeamRelationshipAri
} from "./chunk-5TNKGMMS.mjs";
import "./chunk-35GRSR7B.mjs";
import {
  GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri
} from "./chunk-KGBI7INI.mjs";
import "./chunk-IUR7MTIZ.mjs";
import {
  GraphFeatureFlagHistoryAri
} from "./chunk-JKASSBLA.mjs";
import "./chunk-DFRD6YBM.mjs";
import {
  GraphFeatureFlagAri
} from "./chunk-QAMXFOOY.mjs";
import "./chunk-LDUD7O2G.mjs";
import {
  GraphIncidentHistoryAri
} from "./chunk-RCFEJGPZ.mjs";
import "./chunk-NTBEFMSE.mjs";
import {
  GraphIncidentAri
} from "./chunk-HQJNDY3I.mjs";
import "./chunk-FKF6Z5DH.mjs";
import {
  GraphDesignAri
} from "./chunk-ZP3XNV7G.mjs";
import "./chunk-RBS5RYKM.mjs";
import {
  GraphDevopsComponentAri
} from "./chunk-SFUIVUOI.mjs";
import "./chunk-HLKEKPMY.mjs";
import {
  GraphDocumentHistoryAri
} from "./chunk-FCZKN2Z2.mjs";
import "./chunk-SP4WD2VN.mjs";
import {
  GraphDocumentAri
} from "./chunk-MXZLFSYP.mjs";
import "./chunk-J4TEUSCJ.mjs";
import {
  GraphCommitAri
} from "./chunk-QNGLKTCC.mjs";
import "./chunk-LJ7XURCE.mjs";
import {
  GraphDeploymentHistoryAri
} from "./chunk-VGQR4LCJ.mjs";
import "./chunk-LCOEZ74N.mjs";
import {
  GraphDeploymentAri
} from "./chunk-PNFVJZPS.mjs";
import "./chunk-OON2ZRPC.mjs";
import {
  GraphDesignHistoryAri
} from "./chunk-WDDSFR5N.mjs";
import "./chunk-MTSTC2DF.mjs";
import {
  GraphBranchHistoryAri
} from "./chunk-Y47FNT3K.mjs";
import "./chunk-RL6RLD25.mjs";
import {
  GraphBranchAri
} from "./chunk-WICQAGHE.mjs";
import "./chunk-JJHDDJHM.mjs";
import {
  GraphBuildHistoryAri
} from "./chunk-SFT4TDC2.mjs";
import "./chunk-Q4A3TDFH.mjs";
import {
  GraphBuildAri
} from "./chunk-CZWIZTK4.mjs";
import "./chunk-5YTWJ4WK.mjs";
import "./chunk-3DLTQ4XM.mjs";
import {
  GoogleFormAri
} from "./chunk-OSXG3I5U.mjs";
import "./chunk-X63IP3OI.mjs";
import {
  GooglePresentationAri
} from "./chunk-YTW6GDTJ.mjs";
import "./chunk-5B6S6SWP.mjs";
import {
  GoogleSpreadsheetAri
} from "./chunk-W53MXCPR.mjs";
import "./chunk-466Z3YAU.mjs";
import "./chunk-ZMVXG3Z6.mjs";
import {
  ExusExternalUserAri
} from "./chunk-Y4JMBQPK.mjs";
import "./chunk-IWWNXWBX.mjs";
import "./chunk-5PRHJDN3.mjs";
import {
  FigmaFileAri
} from "./chunk-S4FTMSBF.mjs";
import "./chunk-4R4BPJCR.mjs";
import {
  GoogleDocumentAri
} from "./chunk-IZTWBU2E.mjs";
import "./chunk-5VDZQSQU.mjs";
import "./chunk-YHZUH32W.mjs";
import {
  EcosystemExtensionAri
} from "./chunk-ZZGUWMPL.mjs";
import "./chunk-Q6OH5L4G.mjs";
import "./chunk-J6TNPU7E.mjs";
import {
  ElementsCustomEmojiMetadataAri
} from "./chunk-PKWC5XTY.mjs";
import "./chunk-INZ6GYXP.mjs";
import {
  ElementsWorkspaceAri
} from "./chunk-WGH5QXKA.mjs";
import "./chunk-5TZRCHUN.mjs";
import {
  EcosystemAppAri
} from "./chunk-PIURRD4I.mjs";
import "./chunk-ZSCWAZK4.mjs";
import {
  EcosystemConnectAppAri
} from "./chunk-FGHZK7B7.mjs";
import "./chunk-GEYJT4UH.mjs";
import {
  EcosystemEnvironmentAri
} from "./chunk-BYK6RZCQ.mjs";
import "./chunk-UJZYZKMO.mjs";
import {
  EcosystemExtensionGroupAri
} from "./chunk-YV263EIF.mjs";
import "./chunk-NYRDQJIQ.mjs";
import {
  EcosystemAppEnvironmentVersionAri
} from "./chunk-BSXPXC6Q.mjs";
import "./chunk-CD56DLNS.mjs";
import "./chunk-Q676EFQB.mjs";
import {
  DataLakeLakeConnectionAri
} from "./chunk-AAWH7BK3.mjs";
import "./chunk-2Y4LARQO.mjs";
import "./chunk-PTM25UW5.mjs";
import {
  DevopsProviderAri
} from "./chunk-WISTZ2YA.mjs";
import "./chunk-OILGX3NZ.mjs";
import {
  DevopsRelationshipAri
} from "./chunk-WOCV6R7I.mjs";
import "./chunk-PPXOGRS2.mjs";
import {
  DevopsToolAri
} from "./chunk-AMBFJCLM.mjs";
import "./chunk-WECRO6DG.mjs";
import {
  DevopsContainerAri
} from "./chunk-XOPZUUOS.mjs";
import "./chunk-FGEFELP3.mjs";
import {
  DevopsNamespaceAri
} from "./chunk-XKP2NVAO.mjs";
import "./chunk-H53KN2GD.mjs";
import "./chunk-PQ7YGB4H.mjs";
import {
  ConversationalHelpConversationMessageAri
} from "./chunk-U7LIUFQG.mjs";
import "./chunk-W6PEKGFC.mjs";
import {
  ConversationalHelpConversationAri
} from "./chunk-MGSOKB6Z.mjs";
import "./chunk-YWVV6JNH.mjs";
import "./chunk-FHWJA5QH.mjs";
import {
  ConfluenceWorkspaceAri
} from "./chunk-W6GOYRAF.mjs";
import "./chunk-2BXLBFIW.mjs";
import {
  ConfluenceTaskAri
} from "./chunk-CP5U5JLP.mjs";
import "./chunk-SHZM4TTX.mjs";
import {
  ConfluenceTeamCalendarAri
} from "./chunk-TQGW7FZ5.mjs";
import "./chunk-KKW3GIW7.mjs";
import {
  ConfluenceUserPropertyAri
} from "./chunk-II43EFVQ.mjs";
import "./chunk-NW7N5UEI.mjs";
import {
  ConfluenceWhiteboardAri
} from "./chunk-O4ZBXODC.mjs";
import "./chunk-SRDRWGU4.mjs";
import {
  ConfluencePublicLinkAri
} from "./chunk-3DORSSHF.mjs";
import "./chunk-HXAMB5JS.mjs";
import {
  ConfluenceRoleAri
} from "./chunk-ZWTIQMO3.mjs";
import "./chunk-VTWZOXAB.mjs";
import {
  ConfluenceSiteAri
} from "./chunk-DYUXN4GE.mjs";
import "./chunk-PVOLGFB2.mjs";
import {
  ConfluenceSpaceAri
} from "./chunk-4CHGYGSN.mjs";
import "./chunk-UOPZEJBG.mjs";
import {
  ConfluenceEventAri
} from "./chunk-KULLAOVV.mjs";
import "./chunk-GWVYXALY.mjs";
import {
  ConfluenceLongRunningTaskAri
} from "./chunk-A5ACXR5L.mjs";
import "./chunk-E6K4JCID.mjs";
import {
  ConfluenceNoteAri
} from "./chunk-V2VYC2IL.mjs";
import "./chunk-ZWIRILL2.mjs";
import {
  ConfluencePageAri
} from "./chunk-W3E7IAZG.mjs";
import "./chunk-IMF5GHHQ.mjs";
import {
  ConfluenceCommentAri
} from "./chunk-UNWROYPB.mjs";
import "./chunk-ARFRFJQD.mjs";
import {
  ConfluenceContentAri
} from "./chunk-EGHZXJUH.mjs";
import "./chunk-LVQBHR63.mjs";
import {
  ConfluenceDatabaseAri
} from "./chunk-K3XQ6ARW.mjs";
import "./chunk-PGMODI5S.mjs";
import {
  ConfluenceEmbedAri
} from "./chunk-3Q35HNCF.mjs";
import "./chunk-N5UKNFK5.mjs";
import {
  ConfluenceAdminAnnouncementBannerAri
} from "./chunk-B2O5QAAE.mjs";
import "./chunk-IH6QYLXI.mjs";
import {
  ConfluenceAttachmentAri
} from "./chunk-2EZSRZS5.mjs";
import "./chunk-QCPR2D6J.mjs";
import {
  ConfluenceBlogpostAri
} from "./chunk-DD5H43XA.mjs";
import "./chunk-J5SGYOAJ.mjs";
import "./chunk-BEV3P7ZD.mjs";
import {
  CompassWorkspaceAri
} from "./chunk-T3RVZ2MI.mjs";
import "./chunk-QQTPW4TZ.mjs";
import {
  CompassScorecardAri
} from "./chunk-PP6M6T7Z.mjs";
import "./chunk-2TSZWIWR.mjs";
import {
  CompassSiteAri
} from "./chunk-BI727EY7.mjs";
import "./chunk-SOJW4UXZ.mjs";
import {
  CompassUserDefinedParameterAri
} from "./chunk-6EMOVAGF.mjs";
import "./chunk-U7N5T7JC.mjs";
import {
  CompassWebhookAri
} from "./chunk-YMAGFM2N.mjs";
import "./chunk-RX4T5I6M.mjs";
import {
  CompassMetricDefinitionAri
} from "./chunk-EIT5QOEU.mjs";
import "./chunk-BS6YPB3T.mjs";
import {
  CompassMetricSourceAri
} from "./chunk-F2TUCF6I.mjs";
import "./chunk-H63VXJES.mjs";
import {
  CompassRoleAri
} from "./chunk-YMX57E52.mjs";
import "./chunk-FJ7KFVKZ.mjs";
import {
  CompassComponentAri
} from "./chunk-JX5G7BNF.mjs";
import "./chunk-XW2UZKUO.mjs";
import {
  CompassCustomFieldDefinitionAri
} from "./chunk-3YP4NBJ6.mjs";
import "./chunk-K3BVQB6G.mjs";
import {
  CompassEventSourceAri
} from "./chunk-DZ6RADUJ.mjs";
import "./chunk-G6Y66A5E.mjs";
import {
  CompassIncomingWebhookAri
} from "./chunk-VTNB6S2N.mjs";
import "./chunk-6S4G5T3V.mjs";
import {
  CompassAssistantAnswerAri
} from "./chunk-VX3624BI.mjs";
import "./chunk-VAWIKJGT.mjs";
import {
  CompassComponentLabelAri
} from "./chunk-UQHNK7H4.mjs";
import "./chunk-QOMYCLGO.mjs";
import {
  CompassComponentLinkAri
} from "./chunk-TYBHG6DV.mjs";
import "./chunk-RUUZK374.mjs";
import "./chunk-NWYLOPBR.mjs";
import {
  CommerceWorkbenchResourceAri
} from "./chunk-KH7MR6GL.mjs";
import "./chunk-CVI37ASZ.mjs";
import {
  CommerceQuoteAri
} from "./chunk-FMATMKNJ.mjs";
import "./chunk-2PBMRYJM.mjs";
import {
  CommerceRoleAri
} from "./chunk-3FN3PD6U.mjs";
import "./chunk-T3FMJP7C.mjs";
import {
  CommerceShipToPartyAri
} from "./chunk-PWKDRZAN.mjs";
import "./chunk-MR7K6E3K.mjs";
import {
  CommerceTransactionAccountAri
} from "./chunk-U2LJT766.mjs";
import "./chunk-IG6GXX55.mjs";
import {
  CommercePaymentMethodAri
} from "./chunk-WUYPRJ7C.mjs";
import "./chunk-N3JGQE6H.mjs";
import {
  CommercePricingPlanAri
} from "./chunk-VFVFAYQN.mjs";
import "./chunk-66TMLCVM.mjs";
import {
  CommerceProductAri
} from "./chunk-GA5U5OZM.mjs";
import "./chunk-7KRTMBHT.mjs";
import {
  CommercePromotionAri
} from "./chunk-SXJPHXS4.mjs";
import "./chunk-NDIFFXAM.mjs";
import {
  CommerceInvoiceGroupAri
} from "./chunk-BAFPU5NU.mjs";
import "./chunk-MX6GMSIL.mjs";
import {
  CommerceOfferingRelationshipTemplateAri
} from "./chunk-LB5HFPLJ.mjs";
import "./chunk-2P7D7BWD.mjs";
import {
  CommerceOfferingAri
} from "./chunk-XMCOB5LW.mjs";
import "./chunk-ANF2XXE7.mjs";
import {
  CommerceOrderAri
} from "./chunk-L6K2FJ65.mjs";
import "./chunk-3AXUCZXV.mjs";
import {
  CommerceChangeAri
} from "./chunk-IE5TDG3T.mjs";
import "./chunk-R3MD4VCL.mjs";
import {
  CommerceEntitlementTemplateAri
} from "./chunk-6RT4FQ6G.mjs";
import "./chunk-32JDQQPU.mjs";
import {
  CommerceEntitlementAri
} from "./chunk-VS7SDUYP.mjs";
import "./chunk-NINDB5F3.mjs";
import {
  CommerceInitiativeAri
} from "./chunk-GKUSUZS2.mjs";
import "./chunk-NEKFBJUG.mjs";
import {
  CommerceCatalogAccountAri
} from "./chunk-XSSGYYXG.mjs";
import "./chunk-3FQB6Q73.mjs";
import "./chunk-7USM4R6S.mjs";
import {
  CmdbSchemaAri
} from "./chunk-K4J3EI6U.mjs";
import "./chunk-7NF55AHC.mjs";
import {
  CmdbTypeAri
} from "./chunk-7ISQCFSJ.mjs";
import "./chunk-Q5AHPTNW.mjs";
import {
  CmdbWorkspaceAri
} from "./chunk-5RXPM3KW.mjs";
import "./chunk-UQ5PWXBJ.mjs";
import {
  CmdbImportConfigurationAri
} from "./chunk-35VYXXRU.mjs";
import "./chunk-W3AHDHPL.mjs";
import {
  CmdbObjectAri
} from "./chunk-7POKSVUH.mjs";
import "./chunk-BZ6D72WR.mjs";
import {
  CmdbRoleAri
} from "./chunk-2PUG4ZQU.mjs";
import "./chunk-WF26OXZU.mjs";
import "./chunk-KLYBIY73.mjs";
import {
  CanvasDatabaseAri
} from "./chunk-665I3LV5.mjs";
import "./chunk-LRE2PRU7.mjs";
import {
  CanvasSiteAri
} from "./chunk-U5C33RKO.mjs";
import "./chunk-VCVOT2RL.mjs";
import {
  CanvasWorkspaceAri
} from "./chunk-XXTZNWKF.mjs";
import "./chunk-RSJD3TU6.mjs";
import {
  CmdbAttributeAri
} from "./chunk-D22S5VQ3.mjs";
import "./chunk-XAMITGNQ.mjs";
import "./chunk-QKWQ73L2.mjs";
import {
  BrieRoleAri
} from "./chunk-NFGM6WK5.mjs";
import "./chunk-S5IRDM4L.mjs";
import {
  BrieSiteAri
} from "./chunk-MIWBQOGJ.mjs";
import "./chunk-2CDIPTTF.mjs";
import {
  CanvasBoardAri
} from "./chunk-GQVOV64T.mjs";
import "./chunk-R3VA6JMD.mjs";
import "./chunk-CI2TIA3X.mjs";
import {
  BitbucketRepositoryAri
} from "./chunk-Q4236V3U.mjs";
import "./chunk-4KEXHL4E.mjs";
import {
  BitbucketSiteAri
} from "./chunk-OTIXTWMB.mjs";
import "./chunk-NKJWWYMZ.mjs";
import {
  BitbucketTeamAri
} from "./chunk-MT764SCF.mjs";
import "./chunk-O3ROTQWU.mjs";
import {
  BitbucketWorkspaceAri
} from "./chunk-MMWXVQS4.mjs";
import "./chunk-A35PGQYK.mjs";
import {
  BitbucketOauthConsumerAri
} from "./chunk-GC3PI6VB.mjs";
import "./chunk-B6DXOH6V.mjs";
import {
  BitbucketProjectAri
} from "./chunk-2A53EB64.mjs";
import "./chunk-KZCUOZCO.mjs";
import {
  BitbucketPullrequestAri
} from "./chunk-F6QE22PK.mjs";
import "./chunk-7V62G36F.mjs";
import {
  BitbucketRepositoryTransferAri
} from "./chunk-FR4VWOVF.mjs";
import "./chunk-QL4XDR5W.mjs";
import {
  BitbucketAppAri
} from "./chunk-WCMVHHDS.mjs";
import "./chunk-HN5XXUJU.mjs";
import "./chunk-AIEBCD77.mjs";
import {
  BeaconSiteAri
} from "./chunk-USP2QGCG.mjs";
import "./chunk-JQQF6HSU.mjs";
import {
  BeaconSubscriptionAri
} from "./chunk-VSJXFC64.mjs";
import "./chunk-CWWCGBU4.mjs";
import {
  BeaconWorkspaceAri
} from "./chunk-5NUVF656.mjs";
import "./chunk-POXHAMNN.mjs";
import {
  BeaconDetectionSettingAri
} from "./chunk-NFGFMVQ3.mjs";
import "./chunk-XIUYJLCH.mjs";
import {
  BeaconDetectionAri
} from "./chunk-4JOEA23W.mjs";
import "./chunk-L27OOBSM.mjs";
import {
  BeaconRoleAri
} from "./chunk-V2A5VN6B.mjs";
import "./chunk-YRF7LNG2.mjs";
import "./chunk-2L6JHSPV.mjs";
import {
  AvpRoleAri
} from "./chunk-XLZZIWRF.mjs";
import "./chunk-SZNOZQ4T.mjs";
import {
  AvpSiteAri
} from "./chunk-GGJACO6N.mjs";
import "./chunk-GFZFXLJN.mjs";
import {
  BeaconAlertAri
} from "./chunk-F24UGMZL.mjs";
import "./chunk-TJOHF5RH.mjs";
import {
  BeaconCustomDetectionAri
} from "./chunk-VOTEPGBT.mjs";
import "./chunk-6YGOGAWM.mjs";
import {
  AvpDashboardAri
} from "./chunk-S6NEKKWL.mjs";
import "./chunk-L3TDTHN7.mjs";
import {
  AvpDatasourceAri
} from "./chunk-MSIAEMVR.mjs";
import "./chunk-MIBCZ475.mjs";
import "./chunk-WW2ANN25.mjs";
import {
  AvocadoSiteAri
} from "./chunk-6NLXSTXE.mjs";
import "./chunk-3G3BCF2A.mjs";
import {
  AvocadoUserAri
} from "./chunk-3RIOF4EF.mjs";
import "./chunk-3W3PTAEC.mjs";
import {
  AvocadoAnswerAri
} from "./chunk-5D5ZIZGQ.mjs";
import "./chunk-MBGO4Q2M.mjs";
import {
  AvocadoQuestionAri
} from "./chunk-TJ35IKY4.mjs";
import "./chunk-URX7NO2Y.mjs";
import {
  AvocadoRoleAri
} from "./chunk-BD64QQ5B.mjs";
import "./chunk-DKX2RZ2Z.mjs";
import "./chunk-567EEHI7.mjs";
import {
  AtlashubSiteAri
} from "./chunk-QUFFY2SQ.mjs";
import "./chunk-SJUWQEXL.mjs";
import "./chunk-Z7UEZQHZ.mjs";
import {
  AutomationRoleAri
} from "./chunk-L3K6DVL3.mjs";
import "./chunk-QUWGD5ND.mjs";
import {
  AutomationRuleAri
} from "./chunk-WCYD5VXD.mjs";
import "./chunk-U7AZ64SA.mjs";
import {
  Arm
} from "./chunk-RZZGPFKT.mjs";
import {
  AtlashubIdAri
} from "./chunk-JIZB7ZL7.mjs";
import "./chunk-C4MJWUCX.mjs";
import "./chunk-P5EQPJFA.mjs";
import {
  AnalyticsUserAri
} from "./chunk-GSBEOQ4W.mjs";
import "./chunk-77DTJ44V.mjs";
import "./chunk-ZGGM6QS4.mjs";
import {
  ApprovalDeskWorkflowAri
} from "./chunk-X2L4XKKJ.mjs";
import "./chunk-LIEBAJFZ.mjs";
import {
  ApprovalDeskWorkspaceAri
} from "./chunk-6X5MIG3N.mjs";
import "./chunk-YTLJFQ5C.mjs";
import {
  ApprovalDeskDecisionTableAri
} from "./chunk-KXSHHAGO.mjs";
import "./chunk-GTNXGX3Z.mjs";
import {
  ApprovalDeskRoleAri
} from "./chunk-YPPV7KAD.mjs";
import "./chunk-OQIMDPGX.mjs";
import "./chunk-GER6AKWK.mjs";
import {
  AnyAri,
  Ari
} from "./chunk-YJE4XNCL.mjs";
import {
  Ati
} from "./chunk-CKE2IRS3.mjs";
import "./chunk-GAA5ADPB.mjs";
import {
  ValidationError
} from "./chunk-FAZYALZ2.mjs";

// src/core/types/deprecated/resource-owner-enum.ts
var ResourceOwnerEnum = /* @__PURE__ */ ((ResourceOwnerEnum2) => (ResourceOwnerEnum2.Analytics = "analytics", ResourceOwnerEnum2.ApprovalDesk = "approval-desk", ResourceOwnerEnum2.Atlashub = "atlashub", ResourceOwnerEnum2.Automation = "automation", ResourceOwnerEnum2.Avocado = "avocado", ResourceOwnerEnum2.Avp = "avp", ResourceOwnerEnum2.Beacon = "beacon", ResourceOwnerEnum2.Bitbucket = "bitbucket", ResourceOwnerEnum2.Brie = "brie", ResourceOwnerEnum2.Canvas = "canvas", ResourceOwnerEnum2.Cmdb = "cmdb", ResourceOwnerEnum2.Commerce = "commerce", ResourceOwnerEnum2.Compass = "compass", ResourceOwnerEnum2.Confluence = "confluence", ResourceOwnerEnum2.ConversationalHelp = "conversational-help", ResourceOwnerEnum2.DataLake = "data-lake", ResourceOwnerEnum2.Devops = "devops", ResourceOwnerEnum2.Ecosystem = "ecosystem", ResourceOwnerEnum2.Elements = "elements", ResourceOwnerEnum2.ExusExternal = "exus-external", ResourceOwnerEnum2.Figma = "figma", ResourceOwnerEnum2.Google = "google", ResourceOwnerEnum2.Graph = "graph", ResourceOwnerEnum2.Help = "help", ResourceOwnerEnum2.Identity = "identity", ResourceOwnerEnum2.Jira = "jira", ResourceOwnerEnum2.JiraAlign = "jira-align", ResourceOwnerEnum2.JiraCore = "jira-core", ResourceOwnerEnum2.JiraServicedesk = "jira-servicedesk", ResourceOwnerEnum2.JiraSoftware = "jira-software", ResourceOwnerEnum2.LinkingPlatform = "linking-platform", ResourceOwnerEnum2.MakerSpace = "maker-space", ResourceOwnerEnum2.Marketing = "marketing", ResourceOwnerEnum2.Marketplace = "marketplace", ResourceOwnerEnum2.Measurement = "measurement", ResourceOwnerEnum2.Media = "media", ResourceOwnerEnum2.Mercury = "mercury", ResourceOwnerEnum2.Oauth = "oauth", ResourceOwnerEnum2.Opsgenie = "opsgenie", ResourceOwnerEnum2.Papi = "papi", ResourceOwnerEnum2.Passionfruit = "passionfruit", ResourceOwnerEnum2.PeoplePerftool = "people-perftool", ResourceOwnerEnum2.Platform = "platform", ResourceOwnerEnum2.PlatformServices = "platform-services", ResourceOwnerEnum2.Pollinator = "pollinator", ResourceOwnerEnum2.PostOffice = "post-office", ResourceOwnerEnum2.RuntimeAuthClient = "runtime-auth-client", ResourceOwnerEnum2.Search = "search", ResourceOwnerEnum2.Statuspage = "statuspage", ResourceOwnerEnum2.Support = "support", ResourceOwnerEnum2.Teams = "teams", ResourceOwnerEnum2.Townsquare = "townsquare", ResourceOwnerEnum2.Trello = "trello", ResourceOwnerEnum2.UnifiedHelp = "unified-help", ResourceOwnerEnum2.VirtualAgent = "virtual-agent", ResourceOwnerEnum2))(ResourceOwnerEnum || {});

// src/core/types/deprecated/resource-type-enum.ts
var ResourceTypeEnum = /* @__PURE__ */ ((ResourceTypeEnum2) => (ResourceTypeEnum2.AccountLogin = "account-login", ResourceTypeEnum2.AccountSettings = "account-settings", ResourceTypeEnum2.AdminAnnouncementBanner = "admin-announcement-banner", ResourceTypeEnum2.Alert = "alert", ResourceTypeEnum2.AlertRecipientLink = "alert-recipient-link", ResourceTypeEnum2.AnnouncementBanner = "announcement-banner", ResourceTypeEnum2.Answer = "answer", ResourceTypeEnum2.Api = "api", ResourceTypeEnum2.ApiRequestMetric = "api-request-metric", ResourceTypeEnum2.App = "app", ResourceTypeEnum2.AppEnvironmentVersion = "app-environment-version", ResourceTypeEnum2.Approval = "approval", ResourceTypeEnum2.Article = "article", ResourceTypeEnum2.AssistantAnswer = "assistant-answer", ResourceTypeEnum2.Attachment = "attachment", ResourceTypeEnum2.Attribute = "attribute", ResourceTypeEnum2.AuthPolicy = "auth-policy", ResourceTypeEnum2.AvailableFields = "available-fields", ResourceTypeEnum2.Blogpost = "blogpost", ResourceTypeEnum2.Board = "board", ResourceTypeEnum2.BoardFeature = "board-feature", ResourceTypeEnum2.BoardIssueList = "board-issue-list", ResourceTypeEnum2.Branch = "branch", ResourceTypeEnum2.BranchHistory = "branch-history", ResourceTypeEnum2.Build = "build", ResourceTypeEnum2.BuildHistory = "build-history", ResourceTypeEnum2.BulkOperationTask = "bulk-operation-task", ResourceTypeEnum2.CalendarEvent = "calendar-event", ResourceTypeEnum2.Calibration = "calibration", ResourceTypeEnum2.CallRouting = "call-routing", ResourceTypeEnum2.CannedResponse = "canned-response", ResourceTypeEnum2.Card = "card", ResourceTypeEnum2.CardColor = "card-color", ResourceTypeEnum2.CardLayout = "card-layout", ResourceTypeEnum2.CardParent = "card-parent", ResourceTypeEnum2.CatalogAccount = "catalog-account", ResourceTypeEnum2.Change = "change", ResourceTypeEnum2.Channel = "channel", ResourceTypeEnum2.Check = "check", ResourceTypeEnum2.ClassificationTag = "classification-tag", ResourceTypeEnum2.Client = "client", ResourceTypeEnum2.Column = "column", ResourceTypeEnum2.Comment = "comment", ResourceTypeEnum2.Commit = "commit", ResourceTypeEnum2.Component = "component", ResourceTypeEnum2.ComponentLabel = "component-label", ResourceTypeEnum2.ComponentLink = "component-link", ResourceTypeEnum2.Configuration = "configuration", ResourceTypeEnum2.ConnectApp = "connect-app", ResourceTypeEnum2.Container = "container", ResourceTypeEnum2.Content = "content", ResourceTypeEnum2.Conversation = "conversation", ResourceTypeEnum2.ConversationMessage = "conversation-message", ResourceTypeEnum2.CustomDetection = "custom-detection", ResourceTypeEnum2.CustomEmojiMetadata = "custom-emoji-metadata", ResourceTypeEnum2.Customer = "customer", ResourceTypeEnum2.CustomerDirectory = "customer-directory", ResourceTypeEnum2.CustomerDomain = "customer-domain", ResourceTypeEnum2.CustomerOrganization = "customer-organization", ResourceTypeEnum2.CustomFieldDefinition = "custom-field-definition", ResourceTypeEnum2.CustomFilter = "custom-filter", ResourceTypeEnum2.CustomRole = "custom-role", ResourceTypeEnum2.Dashboard = "dashboard", ResourceTypeEnum2.Database = "database", ResourceTypeEnum2.Datasource = "datasource", ResourceTypeEnum2.DecisionTable = "decision-table", ResourceTypeEnum2.Deployment = "deployment", ResourceTypeEnum2.DeploymentHistory = "deployment-history", ResourceTypeEnum2.DeploymentServiceLink = "deployment-service-link", ResourceTypeEnum2.Design = "design", ResourceTypeEnum2.DesignHistory = "design-history", ResourceTypeEnum2.Detection = "detection", ResourceTypeEnum2.DetectionSetting = "detection-setting", ResourceTypeEnum2.DevopsComponent = "devops-component", ResourceTypeEnum2.Directory = "directory", ResourceTypeEnum2.Document = "document", ResourceTypeEnum2.DocumentHistory = "document-history", ResourceTypeEnum2.Domain = "domain", ResourceTypeEnum2.EmailUuid = "email-uuid", ResourceTypeEnum2.Embed = "embed", ResourceTypeEnum2.Enterprise = "enterprise", ResourceTypeEnum2.Entitlement = "entitlement", ResourceTypeEnum2.EntitlementTemplate = "entitlement-template", ResourceTypeEnum2.EntityProperty = "entity-property", ResourceTypeEnum2.Environment = "environment", ResourceTypeEnum2.Epic = "epic", ResourceTypeEnum2.Escalation = "escalation", ResourceTypeEnum2.Event = "event", ResourceTypeEnum2.EventSource = "event-source", ResourceTypeEnum2.Extension = "extension", ResourceTypeEnum2.ExtensionGroup = "extension-group", ResourceTypeEnum2.Favourite = "favourite", ResourceTypeEnum2.FeatureFlag = "feature-flag", ResourceTypeEnum2.FeatureFlagHistory = "feature-flag-history", ResourceTypeEnum2.Feedback = "feedback", ResourceTypeEnum2.File = "file", ResourceTypeEnum2.Filter = "filter", ResourceTypeEnum2.FilterEmailSubscription = "filter-email-subscription", ResourceTypeEnum2.FlowEditor = "flow-editor", ResourceTypeEnum2.Form = "form", ResourceTypeEnum2.Goal = "goal", ResourceTypeEnum2.Group = "group", ResourceTypeEnum2.HelpCenter = "help-center", ResourceTypeEnum2.HelpDesk = "help-desk", ResourceTypeEnum2.HelpPointer = "help-pointer", ResourceTypeEnum2.Id = "id", ResourceTypeEnum2.ImportConfiguration = "import-configuration", ResourceTypeEnum2.Incident = "incident", ResourceTypeEnum2.IncidentAlertLink = "incident-alert-link", ResourceTypeEnum2.IncidentHistory = "incident-history", ResourceTypeEnum2.IncidentStatusUpdate = "incident-status-update", ResourceTypeEnum2.IncomingCallHistory = "incoming-call-history", ResourceTypeEnum2.IncomingWebhook = "incoming-webhook", ResourceTypeEnum2.Initiative = "initiative", ResourceTypeEnum2.Instance = "instance", ResourceTypeEnum2.Integration = "integration", ResourceTypeEnum2.IntentProjection = "intent-projection", ResourceTypeEnum2.IntentQuestionProjection = "intent-question-projection", ResourceTypeEnum2.IntentRuleProjection = "intent-rule-projection", ResourceTypeEnum2.IntentTemplate = "intent-template", ResourceTypeEnum2.InvoiceGroup = "invoice-group", ResourceTypeEnum2.Issue = "issue", ResourceTypeEnum2.IssueAttachment = "issue-attachment", ResourceTypeEnum2.IssueComment = "issue-comment", ResourceTypeEnum2.IssueFieldMetadata = "issue-field-metadata", ResourceTypeEnum2.IssueFieldOption = "issue-field-option", ResourceTypeEnum2.Issuefieldvalue = "issuefieldvalue", ResourceTypeEnum2.IssueHistory = "issue-history", ResourceTypeEnum2.IssueLink = "issue-link", ResourceTypeEnum2.IssueLinkType = "issue-link-type", ResourceTypeEnum2.IssueRemoteLink = "issue-remote-link", ResourceTypeEnum2.IssueSearchView = "issue-search-view", ResourceTypeEnum2.IssueStatus = "issue-status", ResourceTypeEnum2.IssueType = "issue-type", ResourceTypeEnum2.IssueTypeScheme = "issue-type-scheme", ResourceTypeEnum2.IssueTypeSchemeMapping = "issue-type-scheme-mapping", ResourceTypeEnum2.IssueWorklog = "issue-worklog", ResourceTypeEnum2.JiraProjectAndDevopsToolRelationship = "jira-project-and-devops-tool-relationship", ResourceTypeEnum2.JiraProjectAndDocumentationSpaceRelationship = "jira-project-and-documentation-space-relationship", ResourceTypeEnum2.JiraProjectAndOperationsTeamRelationship = "jira-project-and-operations-team-relationship", ResourceTypeEnum2.JiraProjectAndVcsRepositoryRelationship = "jira-project-and-vcs-repository-relationship", ResourceTypeEnum2.JiraProjectAndVcsRepositoryRelationshipTempMigration = "jira-project-and-vcs-repository-relationship-temp-migration", ResourceTypeEnum2.LakeConnection = "lake-connection", ResourceTypeEnum2.Layout = "layout", ResourceTypeEnum2.Learning = "learning", ResourceTypeEnum2.LifecycleResource = "lifecycle-resource", ResourceTypeEnum2.LifecycleResourcePackage = "lifecycle-resource-package", ResourceTypeEnum2.LifecycleResourcePackageType = "lifecycle-resource-package-type", ResourceTypeEnum2.List = "list", ResourceTypeEnum2.LongRunningTask = "long-running-task", ResourceTypeEnum2.MessageInstance = "message-instance", ResourceTypeEnum2.MessageTemplate = "message-template", ResourceTypeEnum2.MetricDefinition = "metric-definition", ResourceTypeEnum2.MetricSource = "metric-source", ResourceTypeEnum2.Namespace = "namespace", ResourceTypeEnum2.NavigationItem = "navigation-item", ResourceTypeEnum2.Note = "note", ResourceTypeEnum2.Notification = "notification", ResourceTypeEnum2.NotificationType = "notification-type", ResourceTypeEnum2.NotificationTypeScheme = "notification-type-scheme", ResourceTypeEnum2.NotificationUserPreference = "notification-user-preference", ResourceTypeEnum2.OauthClient = "oauthClient", ResourceTypeEnum2.OauthConsumer = "oauth-consumer", ResourceTypeEnum2.Object = "object", ResourceTypeEnum2.Offering = "offering", ResourceTypeEnum2.OfferingRelationshipTemplate = "offering-relationship-template", ResourceTypeEnum2.OperationsWorkspace = "operations-workspace", ResourceTypeEnum2.Order = "order", ResourceTypeEnum2.Org = "org", ResourceTypeEnum2.Organization = "organization", ResourceTypeEnum2.OrgUser = "org-user", ResourceTypeEnum2.Page = "page", ResourceTypeEnum2.PaymentMethod = "payment-method", ResourceTypeEnum2.Performance = "performance", ResourceTypeEnum2.PermissionScheme = "permission-scheme", ResourceTypeEnum2.Plan = "plan", ResourceTypeEnum2.Portal = "portal", ResourceTypeEnum2.PostIncidentReview = "post-incident-review", ResourceTypeEnum2.PostIncidentReviewHistory = "post-incident-review-history", ResourceTypeEnum2.PostIncidentReviewLink = "post-incident-review-link", ResourceTypeEnum2.Presentation = "presentation", ResourceTypeEnum2.PricingPlan = "pricing-plan", ResourceTypeEnum2.Priority = "priority", ResourceTypeEnum2.Product = "product", ResourceTypeEnum2.Program = "program", ResourceTypeEnum2.ProgramStatusUpdate = "program-status-update", ResourceTypeEnum2.Project = "project", ResourceTypeEnum2.ProjectCategory = "project-category", ResourceTypeEnum2.ProjectFeature = "project-feature", ResourceTypeEnum2.ProjectOverview = "project-overview", ResourceTypeEnum2.ProjectRoleActor = "project-role-actor", ResourceTypeEnum2.ProjectShortcut = "project-shortcut", ResourceTypeEnum2.ProjectType = "project-type", ResourceTypeEnum2.Promotion = "promotion", ResourceTypeEnum2.Provider = "provider", ResourceTypeEnum2.PublicLink = "public-link", ResourceTypeEnum2.Pullrequest = "pullrequest", ResourceTypeEnum2.PullRequest = "pull-request", ResourceTypeEnum2.PullRequestHistory = "pull-request-history", ResourceTypeEnum2.Question = "question", ResourceTypeEnum2.Queue = "queue", ResourceTypeEnum2.Quote = "quote", ResourceTypeEnum2.Relationship = "relationship", ResourceTypeEnum2.RemoteLink = "remote-link", ResourceTypeEnum2.RemoteLinkHistory = "remote-link-history", ResourceTypeEnum2.Repository = "repository", ResourceTypeEnum2.RepositoryTransfer = "repository-transfer", ResourceTypeEnum2.RequestForm = "request-form", ResourceTypeEnum2.RequestType = "request-type", ResourceTypeEnum2.Resolution = "resolution", ResourceTypeEnum2.ResourceUsageMetric = "resource-usage-metric", ResourceTypeEnum2.ResourceUsageRecommendation = "resource-usage-recommendation", ResourceTypeEnum2.Role = "role", ResourceTypeEnum2.Rule = "rule", ResourceTypeEnum2.Schedule = "schedule", ResourceTypeEnum2.ScheduleRotation = "schedule-rotation", ResourceTypeEnum2.Schema = "schema", ResourceTypeEnum2.Scorecard = "scorecard", ResourceTypeEnum2.SecureTunnel = "secure-tunnel", ResourceTypeEnum2.SecurityContainer = "security-container", ResourceTypeEnum2.SecurityLevel = "security-level", ResourceTypeEnum2.SecurityWorkspace = "security-workspace", ResourceTypeEnum2.Service = "service", ResourceTypeEnum2.ServiceAndJiraProjectRelationship = "service-and-jira-project-relationship", ResourceTypeEnum2.ServiceAndOperationsTeamRelationship = "service-and-operations-team-relationship", ResourceTypeEnum2.ServiceAndVcsRepositoryRelationship = "service-and-vcs-repository-relationship", ResourceTypeEnum2.ServiceRelationship = "service-relationship", ResourceTypeEnum2.ShipToParty = "ship-to-party", ResourceTypeEnum2.Site = "site", ResourceTypeEnum2.SiteUser = "site-user", ResourceTypeEnum2.Sla = "sla", ResourceTypeEnum2.Space = "space", ResourceTypeEnum2.Spreadsheet = "spreadsheet", ResourceTypeEnum2.Sprint = "sprint", ResourceTypeEnum2.StreamhubSchema = "streamhub-schema", ResourceTypeEnum2.Subscription = "subscription", ResourceTypeEnum2.Swimlane = "swimlane", ResourceTypeEnum2.Tag = "tag", ResourceTypeEnum2.Task = "task", ResourceTypeEnum2.Team = "team", ResourceTypeEnum2.TeamCalendar = "team-calendar", ResourceTypeEnum2.TeamMember = "team-member", ResourceTypeEnum2.Timeline = "timeline", ResourceTypeEnum2.Tool = "tool", ResourceTypeEnum2.TransactionAccount = "transaction-account", ResourceTypeEnum2.Type = "type", ResourceTypeEnum2.User = "user", ResourceTypeEnum2.Userbase = "userbase", ResourceTypeEnum2.UserBoardConfig = "user-board-config", ResourceTypeEnum2.UserBroadcastMessage = "user-broadcast-message", ResourceTypeEnum2.UserDefinedParameter = "user-defined-parameter", ResourceTypeEnum2.UserGrant = "userGrant", ResourceTypeEnum2.UserProperty = "user-property", ResourceTypeEnum2.Version = "version", ResourceTypeEnum2.VersionApprover = "version-approver", ResourceTypeEnum2.ViewConfig = "view-config", ResourceTypeEnum2.Vulnerability = "vulnerability", ResourceTypeEnum2.VulnerabilityHistory = "vulnerability-history", ResourceTypeEnum2.Webhook = "webhook", ResourceTypeEnum2.Whiteboard = "whiteboard", ResourceTypeEnum2.WorkbenchResource = "workbench-resource", ResourceTypeEnum2.Workflow = "workflow", ResourceTypeEnum2.WorkflowScheme = "workflow-scheme", ResourceTypeEnum2.Worklog = "worklog", ResourceTypeEnum2.Workspace = "workspace", ResourceTypeEnum2))(ResourceTypeEnum || {});
export {
  AnalyticsUserAri,
  AnyAri,
  ApprovalDeskDecisionTableAri,
  ApprovalDeskRoleAri,
  ApprovalDeskWorkflowAri,
  ApprovalDeskWorkspaceAri,
  Ari,
  Arm,
  Ati,
  AtlashubIdAri,
  AtlashubSiteAri,
  AutomationRoleAri,
  AutomationRuleAri,
  AvocadoAnswerAri,
  AvocadoQuestionAri,
  AvocadoRoleAri,
  AvocadoSiteAri,
  AvocadoUserAri,
  AvpDashboardAri,
  AvpDatasourceAri,
  AvpRoleAri,
  AvpSiteAri,
  BeaconAlertAri,
  BeaconCustomDetectionAri,
  BeaconDetectionAri,
  BeaconDetectionSettingAri,
  BeaconRoleAri,
  BeaconSiteAri,
  BeaconSubscriptionAri,
  BeaconWorkspaceAri,
  BitbucketAppAri,
  BitbucketOauthConsumerAri,
  BitbucketProjectAri,
  BitbucketPullrequestAri,
  BitbucketRepositoryAri,
  BitbucketRepositoryTransferAri,
  BitbucketSiteAri,
  BitbucketTeamAri,
  BitbucketWorkspaceAri,
  BrieRoleAri,
  BrieSiteAri,
  CanvasBoardAri,
  CanvasDatabaseAri,
  CanvasSiteAri,
  CanvasWorkspaceAri,
  CmdbAttributeAri,
  CmdbImportConfigurationAri,
  CmdbObjectAri,
  CmdbRoleAri,
  CmdbSchemaAri,
  CmdbTypeAri,
  CmdbWorkspaceAri,
  CommerceCatalogAccountAri,
  CommerceChangeAri,
  CommerceEntitlementAri,
  CommerceEntitlementTemplateAri,
  CommerceInitiativeAri,
  CommerceInvoiceGroupAri,
  CommerceOfferingAri,
  CommerceOfferingRelationshipTemplateAri,
  CommerceOrderAri,
  CommercePaymentMethodAri,
  CommercePricingPlanAri,
  CommerceProductAri,
  CommercePromotionAri,
  CommerceQuoteAri,
  CommerceRoleAri,
  CommerceShipToPartyAri,
  CommerceTransactionAccountAri,
  CommerceWorkbenchResourceAri,
  CompassAssistantAnswerAri,
  CompassComponentAri,
  CompassComponentLabelAri,
  CompassComponentLinkAri,
  CompassCustomFieldDefinitionAri,
  CompassEventSourceAri,
  CompassIncomingWebhookAri,
  CompassMetricDefinitionAri,
  CompassMetricSourceAri,
  CompassRoleAri,
  CompassScorecardAri,
  CompassSiteAri,
  CompassUserDefinedParameterAri,
  CompassWebhookAri,
  CompassWorkspaceAri,
  ConfluenceAdminAnnouncementBannerAri,
  ConfluenceAttachmentAri,
  ConfluenceBlogpostAri,
  ConfluenceCommentAri,
  ConfluenceContentAri,
  ConfluenceDatabaseAri,
  ConfluenceEmbedAri,
  ConfluenceEventAri,
  ConfluenceLongRunningTaskAri,
  ConfluenceNoteAri,
  ConfluencePageAri,
  ConfluencePublicLinkAri,
  ConfluenceRoleAri,
  ConfluenceSiteAri,
  ConfluenceSpaceAri,
  ConfluenceTaskAri,
  ConfluenceTeamCalendarAri,
  ConfluenceUserPropertyAri,
  ConfluenceWhiteboardAri,
  ConfluenceWorkspaceAri,
  ConversationalHelpConversationAri,
  ConversationalHelpConversationMessageAri,
  DataLakeLakeConnectionAri,
  DevopsContainerAri,
  DevopsNamespaceAri,
  DevopsProviderAri,
  DevopsRelationshipAri,
  DevopsToolAri,
  EcosystemAppAri,
  EcosystemAppEnvironmentVersionAri,
  EcosystemConnectAppAri,
  EcosystemEnvironmentAri,
  EcosystemExtensionAri,
  EcosystemExtensionGroupAri,
  ElementsCustomEmojiMetadataAri,
  ElementsWorkspaceAri,
  ExusExternalUserAri,
  FigmaFileAri,
  GoogleDocumentAri,
  GoogleFormAri,
  GooglePresentationAri,
  GoogleSpreadsheetAri,
  GraphBranchAri,
  GraphBranchHistoryAri,
  GraphBuildAri,
  GraphBuildHistoryAri,
  GraphCommitAri,
  GraphDeploymentAri,
  GraphDeploymentHistoryAri,
  GraphDesignAri,
  GraphDesignHistoryAri,
  GraphDevopsComponentAri,
  GraphDocumentAri,
  GraphDocumentHistoryAri,
  GraphFeatureFlagAri,
  GraphFeatureFlagHistoryAri,
  GraphIncidentAri,
  GraphIncidentHistoryAri,
  GraphJiraProjectAndDevopsToolRelationshipAri,
  GraphJiraProjectAndDocumentationSpaceRelationshipAri,
  GraphJiraProjectAndOperationsTeamRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipAri,
  GraphJiraProjectAndVcsRepositoryRelationshipTempMigrationAri,
  GraphOperationsWorkspaceAri,
  GraphPostIncidentReviewAri,
  GraphPostIncidentReviewHistoryAri,
  GraphPullRequestAri,
  GraphPullRequestHistoryAri,
  GraphRemoteLinkAri,
  GraphRemoteLinkHistoryAri,
  GraphRepositoryAri,
  GraphSecurityContainerAri,
  GraphSecurityWorkspaceAri,
  GraphServiceAndJiraProjectRelationshipAri,
  GraphServiceAndOperationsTeamRelationshipAri,
  GraphServiceAndVcsRepositoryRelationshipAri,
  GraphServiceAri,
  GraphServiceRelationshipAri,
  GraphVulnerabilityAri,
  GraphVulnerabilityHistoryAri,
  GraphWorkspaceAri,
  HelpArticleAri,
  HelpChannelAri,
  HelpHelpCenterAri,
  HelpHelpDeskAri,
  HelpLayoutAri,
  HelpPortalAri,
  HelpRequestFormAri,
  IdentityAppAri,
  IdentityAuthPolicyAri,
  IdentityCustomerDirectoryAri,
  IdentityCustomerOrganizationAri,
  IdentityDirectoryAri,
  IdentityDomainAri,
  IdentityGroupAri,
  IdentityOauthClientAri,
  IdentityRoleAri,
  IdentitySiteAri,
  IdentityTeamAri,
  IdentityTeamMemberAri,
  IdentityUserAri,
  IdentityUserGrantAri,
  IdentityUserbaseAri,
  JiraAlignEpicAri,
  JiraAlignInstanceAri,
  JiraAnnouncementBannerAri,
  JiraAvailableFieldsAri,
  JiraBranchAri,
  JiraBranchHistoryAri,
  JiraBuildAri,
  JiraBuildHistoryAri,
  JiraBulkOperationTaskAri,
  JiraCommentAri,
  JiraCommitAri,
  JiraComponentAri,
  JiraConnectAppAri,
  JiraCoreRoleAri,
  JiraCoreSiteAri,
  JiraDashboardAri,
  JiraDeploymentAri,
  JiraDeploymentHistoryAri,
  JiraDesignAri,
  JiraDesignHistoryAri,
  JiraDevopsComponentAri,
  JiraDocumentAri,
  JiraDocumentHistoryAri,
  JiraEntityPropertyAri,
  JiraFavouriteAri,
  JiraFeatureFlagAri,
  JiraFeatureFlagHistoryAri,
  JiraFilterAri,
  JiraFilterEmailSubscriptionAri,
  JiraIncidentAri,
  JiraIncidentHistoryAri,
  JiraIssueAri,
  JiraIssueAttachmentAri,
  JiraIssueCommentAri,
  JiraIssueFieldMetadataAri,
  JiraIssueFieldOptionAri,
  JiraIssueHistoryAri,
  JiraIssueLinkAri,
  JiraIssueLinkTypeAri,
  JiraIssueRemoteLinkAri,
  JiraIssueSearchViewAri,
  JiraIssueStatusAri,
  JiraIssueTypeAri,
  JiraIssueTypeSchemeAri,
  JiraIssueTypeSchemeMappingAri,
  JiraIssueWorklogAri,
  JiraIssuefieldvalueAri,
  JiraNavigationItemAri,
  JiraNotificationTypeAri,
  JiraNotificationTypeSchemeAri,
  JiraNotificationUserPreferenceAri,
  JiraOperationsWorkspaceAri,
  JiraPermissionSchemeAri,
  JiraPlanAri,
  JiraPostIncidentReviewAri,
  JiraPostIncidentReviewHistoryAri,
  JiraPostIncidentReviewLinkAri,
  JiraPriorityAri,
  JiraProductAri,
  JiraProjectAri,
  JiraProjectCategoryAri,
  JiraProjectFeatureAri,
  JiraProjectOverviewAri,
  JiraProjectRoleActorAri,
  JiraProjectShortcutAri,
  JiraProjectTypeAri,
  JiraPullRequestAri,
  JiraPullRequestHistoryAri,
  JiraRemoteLinkAri,
  JiraRemoteLinkHistoryAri,
  JiraRepositoryAri,
  JiraResolutionAri,
  JiraResourceUsageMetricAri,
  JiraResourceUsageRecommendationAri,
  JiraRoleAri,
  JiraSecurityContainerAri,
  JiraSecurityLevelAri,
  JiraSecurityWorkspaceAri,
  JiraServicedeskApprovalAri,
  JiraServicedeskCalendarEventAri,
  JiraServicedeskCannedResponseAri,
  JiraServicedeskOrganizationAri,
  JiraServicedeskQueueAri,
  JiraServicedeskRequestTypeAri,
  JiraServicedeskRoleAri,
  JiraServicedeskSiteAri,
  JiraServicedeskSlaAri,
  JiraSiteAri,
  JiraSoftwareBoardAri,
  JiraSoftwareBoardFeatureAri,
  JiraSoftwareBoardIssueListAri,
  JiraSoftwareCardAri,
  JiraSoftwareCardColorAri,
  JiraSoftwareCardLayoutAri,
  JiraSoftwareCardParentAri,
  JiraSoftwareColumnAri,
  JiraSoftwareCustomFilterAri,
  JiraSoftwareRoleAri,
  JiraSoftwareSiteAri,
  JiraSoftwareSwimlaneAri,
  JiraSoftwareUserBoardConfigAri,
  JiraSoftwareViewConfigAri,
  JiraSprintAri,
  JiraUserBroadcastMessageAri,
  JiraVersionApproverAri,
  JiraVersionAri,
  JiraVulnerabilityAri,
  JiraVulnerabilityHistoryAri,
  JiraWorkflowAri,
  JiraWorkflowSchemeAri,
  JiraWorklogAri,
  JiraWorkspaceAri,
  LinkingPlatformDatasourceAri,
  MakerSpaceGroupAri,
  MarketingCustomerDomainAri,
  MarketplaceTeamAri,
  MeasurementEmailUuidAri,
  MeasurementSiteUserAri,
  MeasurementUserAri,
  MediaFileAri,
  MercuryCommentAri,
  MercuryProgramAri,
  MercuryProgramStatusUpdateAri,
  MercuryRoleAri,
  MercurySiteAri,
  OauthClientAri,
  OpsgenieAccountLoginAri,
  OpsgenieAccountSettingsAri,
  OpsgenieAlertAri,
  OpsgenieAlertRecipientLinkAri,
  OpsgenieApiRequestMetricAri,
  OpsgenieCallRoutingAri,
  OpsgenieCustomRoleAri,
  OpsgenieDeploymentAri,
  OpsgenieDeploymentServiceLinkAri,
  OpsgenieEscalationAri,
  OpsgenieEventAri,
  OpsgenieIncidentAlertLinkAri,
  OpsgenieIncidentAri,
  OpsgenieIncidentStatusUpdateAri,
  OpsgenieIncomingCallHistoryAri,
  OpsgenieIntegrationAri,
  OpsgenieNotificationAri,
  OpsgenieRoleAri,
  OpsgenieScheduleAri,
  OpsgenieScheduleRotationAri,
  OpsgenieSiteAri,
  OpsgenieTeamAri,
  OpsgenieTimelineAri,
  OpsgenieWorkspaceAri,
  PapiApiAri,
  PapiRoleAri,
  PassionfruitUserAri,
  PeoplePerftoolCalibrationAri,
  PeoplePerftoolFeedbackAri,
  PeoplePerftoolPerformanceAri,
  PlatformClassificationTagAri,
  PlatformLifecycleResourceAri,
  PlatformLifecycleResourcePackageAri,
  PlatformLifecycleResourcePackageTypeAri,
  PlatformOrgAri,
  PlatformOrgUserAri,
  PlatformSecureTunnelAri,
  PlatformServicesStreamhubSchemaAri,
  PlatformSiteAri,
  PollinatorCheckAri,
  PostOfficeMessageInstanceAri,
  PostOfficeMessageTemplateAri,
  ResourceOwnerEnum as ResourceOwner,
  ResourceTypeEnum as ResourceType,
  RuntimeAuthClientRoleAri,
  SearchEventAri,
  StatuspageRoleAri,
  StatuspageSiteAri,
  SupportCustomerAri,
  TeamsTeamAri,
  TownsquareCommentAri,
  TownsquareGoalAri,
  TownsquareHelpPointerAri,
  TownsquareLearningAri,
  TownsquareProjectAri,
  TownsquareQuestionAri,
  TownsquareRoleAri,
  TownsquareSiteAri,
  TownsquareTagAri,
  TrelloBoardAri,
  TrelloCardAri,
  TrelloEnterpriseAri,
  TrelloListAri,
  TrelloRoleAri,
  TrelloSiteAri,
  TrelloUserAri,
  TrelloWorkspaceAri,
  UnifiedHelpRoleAri,
  UnifiedHelpSiteAri,
  UnifiedHelpWorkspaceAri,
  ValidationError,
  VirtualAgentConfigurationAri,
  VirtualAgentConversationAri,
  VirtualAgentFlowEditorAri,
  VirtualAgentIntentProjectionAri,
  VirtualAgentIntentQuestionProjectionAri,
  VirtualAgentIntentRuleProjectionAri,
  VirtualAgentIntentTemplateAri
};
