// @atlaskit/range/dist/cjs/theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transitionDuration = exports.track = exports.thumb = exports.input = void 0;
var _colors = require("@atlaskit/theme/colors");
var transitionDuration = exports.transitionDuration = '0.2s';
var input = exports.input = {
  height: 40
};
var thumb = exports.thumb = {
  size: 16,
  borderWidth: 2,
  background: {
    default: "var(--ds-background-neutral-bold, ".concat(_colors.B400, ")"),
    hovered: "var(--ds-background-neutral-bold-hovered, ".concat(_colors.B300, ")"),
    pressed: "var(--ds-background-neutral-bold-pressed, ".concat(_colors.B200, ")")
  },
  borderColor: {
    default: 'transparent',
    focused: "var(--ds-border-focused, ".concat(_colors.B200, ")")
  },
  boxShadow: {
    default: "var(--ds-UNSAFE-transparent, ".concat("0 4px 8px -2px ".concat(_colors.N50A, ", 0 0 1px ").concat(_colors.N60A), ")"),
    disabled: "var(--ds-shadow-raised, ".concat("0 0 1px ".concat(_colors.N60A), ")")
  }
};
var track = exports.track = {
  height: 4,
  /**
   * borderRadius >= height / 2 to create a pill shape.
   * Using '50%' creates an ellipse.
   */
  borderRadius: 2,
  background: {
    default: "var(--ds-background-neutral, ".concat(_colors.N30, ")"),
    hovered: "var(--ds-background-neutral-hovered, ".concat(_colors.N40, ")")
  },
  foreground: {
    default: "var(--ds-background-neutral-bold, ".concat(_colors.B400, ")"),
    hovered: "var(--ds-background-neutral-bold-hovered, ".concat(_colors.B300, ")")
  }
};
