// @atlaskit/dynamic-table/dist/cjs/styled/constants.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.truncationWidthStyles = exports.overflowTruncateStyles = exports.getTruncationStyleVars = exports.fixedSizeTruncateStyles = exports.cellStyles = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("@emotion/react");
// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var CSS_VAR_WIDTH = '--local-dynamic-table-width';
// eslint-disable-next-line @atlaskit/design-system/no-exported-css, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var truncationWidthStyles = exports.truncationWidthStyles = (0, _react.css)({
  width: "var(".concat(CSS_VAR_WIDTH, ")")
});

// eslint-disable-next-line @atlaskit/design-system/no-exported-css, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var fixedSizeTruncateStyles = exports.fixedSizeTruncateStyles = (0, _react.css)({
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
});

// eslint-disable-next-line @atlaskit/design-system/no-exported-css, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var overflowTruncateStyles = exports.overflowTruncateStyles = (0, _react.css)({
  overflow: 'hidden'
});
var getTruncationStyleVars = exports.getTruncationStyleVars = function getTruncationStyleVars(_ref) {
  var width = _ref.width;
  return typeof width !== 'undefined' ? (0, _defineProperty2.default)({}, CSS_VAR_WIDTH, "".concat(width, "%")) : undefined;
};

// eslint-disable-next-line @atlaskit/design-system/no-exported-css, @atlaskit/ui-styling-standard/no-exported-styles -- Ignored via go/DSP-18766
var cellStyles = exports.cellStyles = (0, _react.css)({
  padding: "var(--ds-space-050, 4px)".concat(" ", "var(--ds-space-100, 8px)"),
  border: 'none',
  textAlign: 'left',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
  '&:first-of-type': {
    paddingInlineStart: "var(--ds-space-0, 0px)"
  },
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-unsafe-selectors -- Ignored via go/DSP-18766
  '&:last-child': {
    paddingInlineEnd: "var(--ds-space-0, 0px)"
  }
});
