// @atlaskit/code/dist/cjs/internal/theme/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VAR_CODE_LINE_NUMBER_BG_COLOR = exports.VAR_CODE_BG_COLOR = exports.SPACING = exports.LINE_NUMBER_GUTTER = exports.HIGHLIGHT_BORDER_WIDTH = exports.CODE_LINE_HEIGHT = exports.CODE_FONT_SIZE = exports.CODE_BLOCK_SELECTOR = void 0;
var _constants = require("@atlaskit/theme/constants");
// eslint-disable-next-line @atlaskit/design-system/no-deprecated-imports

var CODE_FONT_SIZE = exports.CODE_FONT_SIZE = 12;
var CODE_LINE_HEIGHT = exports.CODE_LINE_HEIGHT = '20px';
var HIGHLIGHT_BORDER_WIDTH = exports.HIGHLIGHT_BORDER_WIDTH = '4px';
var SPACING = exports.SPACING = (0, _constants.gridSize)();
var LINE_NUMBER_GUTTER = exports.LINE_NUMBER_GUTTER = SPACING * 2;
var VAR_CODE_LINE_NUMBER_BG_COLOR = exports.VAR_CODE_LINE_NUMBER_BG_COLOR = '--ds--code--line-number-bg-color';
var VAR_CODE_BG_COLOR = exports.VAR_CODE_BG_COLOR = '--ds--code--bg-color';

// selector for codeblock
var CODE_BLOCK_SELECTOR = exports.CODE_BLOCK_SELECTOR = "data-ds--code--code-block";
