// @atlaskit/progress-tracker/dist/cjs/internal/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.varTransitionSpeed = exports.varTransitionEasing = exports.varTransitionDelay = exports.varSpacing = exports.varMarkerColor = exports.varBackgroundColor = exports.TRANSITION_SPEED = exports.PROGRESS_BAR_HEIGHT = exports.LINEAR_TRANSITION_SPEED = exports.LABEL_TOP_SPACING = exports.HALF_GRID_SIZE = exports.ANIMATION_EASE_OUT = void 0;
var TRANSITION_SPEED = exports.TRANSITION_SPEED = 300;
var LINEAR_TRANSITION_SPEED = exports.LINEAR_TRANSITION_SPEED = 50;
var ANIMATION_EASE_OUT = exports.ANIMATION_EASE_OUT = 'cubic-bezier(0.15,1,0.3,1)';
var varSpacing = exports.varSpacing = '--ds--pt--sp';
var varTransitionSpeed = exports.varTransitionSpeed = '--ds--pt--ts';
var varTransitionDelay = exports.varTransitionDelay = '--ds--pt--td';
var varTransitionEasing = exports.varTransitionEasing = '--ds--pt--te';
var varMarkerColor = exports.varMarkerColor = '--ds--pt--mc';
var varBackgroundColor = exports.varBackgroundColor = '--ds--pt--bg';
var HALF_GRID_SIZE = exports.HALF_GRID_SIZE = "var(--ds-space-050, 4px)";
var PROGRESS_BAR_HEIGHT = exports.PROGRESS_BAR_HEIGHT = "var(--ds-space-100, 8px)";

// Labels sit 16px from bottom of progress bar
// i.e. 8 (bar height) + 16 - 4 (line height 24->20 in PYX-695) = 20px
var LABEL_TOP_SPACING = exports.LABEL_TOP_SPACING = "var(--ds-space-250, 20px)";
