// @atlaskit/motion/dist/es2019/entering/zoom-in.js
import React from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

import KeyframesMotion from './keyframes-motion';
export const zoomInAnimation = () => ({
  '0%': {
    opacity: 0,
    transform: 'scale(0.5)'
  },
  '50%': {
    opacity: 1
  },
  '75%': {
    transform: 'scale(1.25)'
  },
  '100%': {
    transform: 'scale(1)'
  }
});
export const shrinkOutAnimation = () => ({
  to: {
    opacity: 0,
    transform: 'scale(0.75)'
  }
});

/**
 * __ZoomIn__
 *
 * Will over zoom an element into position.
 *
 * - [Examples](https://atlaskit.atlassian.com/packages/design-system/motion/docs/entering-motions)
 */
const ZoomIn = ({
  children,
  duration = 125,
  isPaused,
  onFinish
}) => {
  return /*#__PURE__*/React.createElement(KeyframesMotion, {
    duration: duration,
    enteringAnimation: zoomInAnimation(),
    exitingAnimation: shrinkOutAnimation(),
    animationTimingFunction: () => 'ease-in-out',
    isPaused: isPaused,
    onFinish: onFinish
  }, children);
};
export default ZoomIn;
