// @atlaskit/lozenge/dist/cjs/theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultTextColor = exports.defaultBackgroundColor = exports.boldTextColor = exports.boldBackgroundColor = void 0;
var _colors = require("@atlaskit/theme/colors");
/* eslint-disable @atlaskit/design-system/ensure-design-token-usage */
// TODO: DSP-1392 this file will be removed, don't import anything from this file elsewhere.

/**
 * @deprecated
 */
var defaultBackgroundColor = exports.defaultBackgroundColor = {
  default: {
    light: _colors.N40,
    dark: _colors.N40
  },
  inprogress: {
    light: _colors.B50,
    dark: _colors.B50
  },
  moved: {
    light: _colors.Y75,
    dark: _colors.Y75
  },
  new: {
    light: _colors.P50,
    dark: _colors.P50
  },
  removed: {
    light: _colors.R50,
    dark: _colors.R50
  },
  success: {
    light: _colors.G50,
    dark: _colors.G50
  }
};

/**
 * @deprecated
 */
var defaultTextColor = exports.defaultTextColor = {
  default: {
    light: _colors.N500,
    dark: _colors.N500
  },
  inprogress: {
    light: _colors.B500,
    dark: _colors.B500
  },
  moved: {
    light: _colors.N800,
    dark: _colors.N800
  },
  new: {
    light: _colors.P500,
    dark: _colors.P500
  },
  removed: {
    light: _colors.R500,
    dark: _colors.R500
  },
  success: {
    light: _colors.G500,
    dark: _colors.G500
  }
};

/**
 * @deprecated
 */
var boldBackgroundColor = exports.boldBackgroundColor = {
  default: {
    light: _colors.N500,
    dark: _colors.N500
  },
  inprogress: {
    light: _colors.B400,
    dark: _colors.B400
  },
  moved: {
    light: _colors.Y500,
    dark: _colors.Y500
  },
  new: {
    light: _colors.P400,
    dark: _colors.P400
  },
  removed: {
    light: _colors.R400,
    dark: _colors.R400
  },
  success: {
    light: _colors.G400,
    dark: _colors.G400
  }
};

/**
 * @deprecated
 */
var boldTextColor = exports.boldTextColor = {
  default: {
    light: _colors.N0,
    dark: _colors.N0
  },
  inprogress: {
    light: _colors.N0,
    dark: _colors.N0
  },
  moved: {
    light: _colors.N800,
    dark: _colors.N800
  },
  new: {
    light: _colors.N0,
    dark: _colors.N0
  },
  removed: {
    light: _colors.N0,
    dark: _colors.N0
  },
  success: {
    light: _colors.N0,
    dark: _colors.N0
  }
};

/**
 * @deprecated
 */
