// @atlaskit/range/dist/es2019/theme.js
import { B200, B300, B400, N30, N40, N50A, N60A } from '@atlaskit/theme/colors';
export const transitionDuration = '0.2s';
export const input = {
  height: 40
};
export const thumb = {
  size: 16,
  borderWidth: 2,
  background: {
    default: `var(--ds-background-neutral-bold, ${B400})`,
    hovered: `var(--ds-background-neutral-bold-hovered, ${B300})`,
    pressed: `var(--ds-background-neutral-bold-pressed, ${B200})`
  },
  borderColor: {
    default: 'transparent',
    focused: `var(--ds-border-focused, ${B200})`
  },
  boxShadow: {
    default: `var(--ds-UNSAFE-transparent, ${`0 4px 8px -2px ${N50A}, 0 0 1px ${N60A}`})`,
    disabled: `var(--ds-shadow-raised, ${`0 0 1px ${N60A}`})`
  }
};
export const track = {
  height: 4,
  /**
   * borderRadius >= height / 2 to create a pill shape.
   * Using '50%' creates an ellipse.
   */
  borderRadius: 2,
  background: {
    default: `var(--ds-background-neutral, ${N30})`,
    hovered: `var(--ds-background-neutral-hovered, ${N40})`
  },
  foreground: {
    default: `var(--ds-background-neutral-bold, ${B400})`,
    hovered: `var(--ds-background-neutral-bold-hovered, ${B300})`
  }
};
