// @atlaskit/dynamic-table/dist/cjs/components/rankable/table-cell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RankableTableCell = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = _interopRequireDefault(require("react"));
var _withDimensions = _interopRequireDefault(require("../../hoc/with-dimensions"));
var _helpers = require("../../internal/helpers");
var _tableCell = require("../../styled/rankable/table-cell");
var _excluded = ["content", "testId"];
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
var RankableTableCell = exports.RankableTableCell = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(RankableTableCell, _React$Component);
  var _super = _createSuper(RankableTableCell);
  function RankableTableCell() {
    (0, _classCallCheck2.default)(this, RankableTableCell);
    return _super.apply(this, arguments);
  }
  (0, _createClass2.default)(RankableTableCell, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
        cell = _this$props.cell,
        head = _this$props.head,
        isFixedSize = _this$props.isFixedSize,
        isRanking = _this$props.isRanking,
        refWidth = _this$props.refWidth,
        innerRef = _this$props.innerRef,
        testId = _this$props.testId;
      var content = cell.content,
        cellTestId = cell.testId,
        restCellProps = (0, _objectWithoutProperties2.default)(cell, _excluded);
      var _ref = head || {},
        shouldTruncate = _ref.shouldTruncate,
        width = _ref.width;
      var inlineStyles = (0, _helpers.inlineStylesIfRanking)(isRanking, refWidth);
      return /*#__PURE__*/_react.default.createElement(_tableCell.RankableTableBodyCell, (0, _extends2.default)({}, restCellProps, {
        isFixedSize: isFixedSize,
        shouldTruncate: shouldTruncate,
        width: width,
        isRanking: isRanking
        // eslint-disable-next-line @atlaskit/ui-styling-standard/enforce-style-prop -- Ignored via go/DSP-18766
        ,
        style: inlineStyles,
        onKeyDown: function onKeyDown(e) {
          return e.stopPropagation();
        },
        innerRef: innerRef,
        "data-testid": cellTestId || testId && "".concat(testId, "--rankable--table--body--cell")
      }), content);
    }
  }]);
  return RankableTableCell;
}(_react.default.Component);
var _default = exports.default = (0, _withDimensions.default)(RankableTableCell);
