// @atlaskit/tag-group/dist/es2019/TagGroup/index.js
/**
 * @jsxRuntime classic
 */
/** @jsx jsx */

import { forwardRef } from 'react';

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766
import { css, jsx } from '@emotion/react';
const baseStyles = css({
  display: 'flex',
  width: '100%',
  flexWrap: 'wrap'
});
const justifyStartStyles = css({
  justifyContent: 'flex-start'
});
const justifyEndStyles = css({
  justifyContent: 'flex-end'
});

/**
 * __Tag group__
 *
 * A tag group controls the layout and alignment for a collection of tags.
 *
 * - [Examples](https://atlassian.design/components/tag-group/examples)
 * - [Code](https://atlassian.design/components/tag-group/code)
 * - [Usage](https://atlassian.design/components/tag-group/usage)
 */
const TagGroup = /*#__PURE__*/forwardRef(({
  alignment = 'start',
  children
}, ref) => {
  return jsx("div", {
    ref: ref,
    css: [baseStyles, alignment === 'start' && justifyStartStyles, alignment === 'end' && justifyEndStyles]
  }, children);
});
export default TagGroup;
