// @atlaskit/tag/dist/cjs/tag/internal/shared/before.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _constants = require("../../../constants");
/**
 * @jsxRuntime classic
 */
/** @jsx jsx */

// eslint-disable-next-line @atlaskit/ui-styling-standard/use-compiled -- Ignored via go/DSP-18766

var beforeElementStyles = (0, _react.css)({
  display: 'flex',
  position: 'absolute',
  alignItems: 'center',
  justifyContent: 'center',
  // eslint-disable-next-line @atlaskit/ui-styling-standard/no-imported-style-values, @atlaskit/ui-styling-standard/no-unsafe-values -- Ignored via go/DSP-18766
  borderRadius: "var(".concat(_constants.cssVar.borderRadius, ")"),
  insetBlockStart: "var(--ds-space-0, 0px)"
});
var Before = function Before(_ref) {
  var elemBefore = _ref.elemBefore;
  return elemBefore ? (0, _react.jsx)("span", {
    css: beforeElementStyles
  }, elemBefore) : null;
};

// eslint-disable-next-line @repo/internal/react/require-jsdoc
var _default = exports.default = Before;
