// @atlaskit/modal-dialog/dist/cjs/internal/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.width = exports.verticalOffset = exports.textColor = exports.padding = exports.keylineHeight = exports.keylineColor = exports.iconColor = exports.gutter = exports.borderRadius = void 0;
var _colors = require("@atlaskit/theme/colors");
var _constants = require("@atlaskit/theme/constants");
var width = exports.width = {
  values: ['small', 'medium', 'large', 'x-large'],
  widths: {
    small: 400,
    medium: 600,
    large: 800,
    'x-large': 968
  },
  defaultValue: 'medium'
};
var gutter = exports.gutter = 60;
var gridSize = (0, _constants.gridSize)();
var borderRadius = exports.borderRadius = (0, _constants.borderRadius)();
var verticalOffset = exports.verticalOffset = gridSize * 2;
var padding = exports.padding = gridSize * 3;
var keylineHeight = exports.keylineHeight = 2;
var keylineColor = exports.keylineColor = "var(--ds-border, ".concat(_colors.N30, ")");
var textColor = exports.textColor = "var(--ds-text, ".concat(_colors.N900, ")");
var iconColor = exports.iconColor = {
  danger: "var(--ds-icon-danger, ".concat(_colors.R400, ")"),
  warning: "var(--ds-icon-warning, #D97008)"
};
